Error encountered while attempting to run `./program.sh 41`

I cloned the repo and ran the `build_docker_container.sh` script from the `linux/` directory then attempting to program a simple [blink example](https://raw.githubusercontent.com/tttapa/Arduino-AVR-CMake/master/blink.cpp) for a Teensy 4.1, but encountered the following error

```
$ ./program.sh 41
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

Sending build context to Docker daemon  158.7kB
Step 1/37 : FROM ubuntu:20.04
 ---> 83a4bf3bb050
Step 2/37 : ARG WORKDIR=/root
 ---> Using cache
 ---> 2088ae0d6f03
Step 3/37 : ENV DOCKER_ARDUINO_VERSION=arduino-1.8.19
 ---> Using cache
 ---> a64f692fe0ef
Step 4/37 : ENV DOCKER_TEENSY_TD_VERSION=td_156
 ---> Using cache
 ---> 47a5dbeecace
Step 5/37 : ENV DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 0db6bea9dd2f
Step 6/37 : RUN apt -y update &&     apt -y install     avr-libc     binutils-avr     cmake     g++     gcc     gcc-avr     git     libfontconfig1     libusb-dev     libxft-dev     make     python3.8     unzip     vim     wget     xz-utils     && apt clean && rm -rf /var/lib/apt/lists
 ---> Using cache
 ---> 52d95ce3b8ee
Step 7/37 : RUN mkdir -p /etc/udev/rules.d/  &&     echo    ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04*", ENV{ID_MM_DEVICE_IGNORE}="1", ENV{ID_MM_PORT_IGNORE}="1"             ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04[789a]*", ENV{MTP_NO_PROBE}="1"             KERNEL=="ttyACM*", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04*", MODE:="0666", RUN:="/bin/stty -F /dev/%k raw -echo"             KERNEL=="hidraw*", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04*", MODE:="0666"             SUBSYSTEMS=="usb", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04*", MODE:="0666" >> /etc/udev/rules.d/00-teensy.rules
 ---> Using cache
 ---> 55655d018e9e
Step 8/37 : WORKDIR /teensyduino
 ---> Using cache
 ---> 1e1ec2ab3bf3
Step 9/37 : RUN wget -q https://downloads.arduino.cc/${DOCKER_ARDUINO_VERSION}-linux64.tar.xz &&     wget -q https://www.pjrc.com/teensy/${DOCKER_TEENSY_TD_VERSION}/TeensyduinoInstall.linux64
 ---> Using cache
 ---> a037ebb3c8ca
Step 10/37 : WORKDIR /teensy_cli
 ---> Using cache
 ---> 0f64ede59e05
Step 11/37 : RUN git clone -q https://github.com/PaulStoffregen/teensy_loader_cli &&     cd teensy_loader_cli &&     make &&     cd rebootor &&     make
 ---> Using cache
 ---> ea238b6a3f9f
Step 12/37 : ENV PATH="$PATH:/teensy_cli/teensy_loader_cli/"
 ---> Using cache
 ---> 5d955a355ef1
Step 13/37 : WORKDIR /teensyduino
 ---> Using cache
 ---> 63b8ca0560ad
Step 14/37 : RUN tar -xf ${DOCKER_ARDUINO_VERSION}-linux64.tar.xz &&     rm ${DOCKER_ARDUINO_VERSION}-linux64.tar.xz &&     chmod 755 TeensyduinoInstall.linux64 &&     ./TeensyduinoInstall.linux64 --dir=/teensyduino/${DOCKER_ARDUINO_VERSION} &&     mkdir -p ./libraries &&     mkdir -p ./cores &&     mkdir -p ./src &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/teensy/avr/libraries/* ./libraries &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/teensy/avr/cores/* ./cores
 ---> Using cache
 ---> e2cdfac9a02d
Step 15/37 : WORKDIR /teensyduino/libraries
 ---> Using cache
 ---> d9b3645a640b
Step 16/37 : RUN      git clone https://github.com/adafruit/Adafruit_BusIO.git &&     git clone https://github.com/adafruit/Adafruit-GFX-Library.git &&     mkdir /src
 ---> Using cache
 ---> 2db85b2ba032
Step 17/37 : WORKDIR /teensyduino/bin
 ---> Using cache
 ---> 787d75edad7b
Step 18/37 : RUN cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/bin/* . &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/lib/gcc/arm-none-eabi/5.4.1/* . &&     cp -u /teensyduino/cores/teensy3/*.ld . &&     cp -u /teensyduino/cores/teensy4/*.ld . &&     mkdir -p /teensyduino/lib/gcc &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/lib/gcc/arm-none-eabi/5.4.1/* /teensyduino/lib/gcc
 ---> Using cache
 ---> 85b0a4e6fb1a
Step 19/37 : WORKDIR /teensyduino/include
 ---> Using cache
 ---> bb851de7420a
Step 20/37 : RUN cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/arm-none-eabi/include/* . &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/arm-none-eabi/lib/armv7e-m/* . &&     cp /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/arm-none-eabi/lib/nano.specs . &&     cp /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/arm-none-eabi/lib/lib* . &&     mv crt0.o /teensyduino/lib/gcc
 ---> Using cache
 ---> e2a249f44e74
Step 21/37 : ENV PATH="/teensyduino/bin:/teensyduino/include:/teensyduino/bin/plugin/include:$PATH"
 ---> Using cache
 ---> bd7e94a29c0d
Step 22/37 : WORKDIR /teensyduino
 ---> Using cache
 ---> 5ecfee83f58d
Step 23/37 : RUN mkdir -p /teensyduino/install &&     mkdir -p /teensyduino/build
 ---> Using cache
 ---> 199d9015efdb
Step 24/37 : ADD internal/CMake/project_root/* /teensyduino/
 ---> Using cache
 ---> 96a57088bf92
Step 25/37 : ADD internal/CMake/toolchain/* /teensyduino/
 ---> Using cache
 ---> 10a49ba9c867
Step 26/37 : ADD config.json /teensyduino/
 ---> Using cache
 ---> 539ab76bc6ae
Step 27/37 : ADD internal/CMake/select_all/CMakeLists.txt /teensyduino/cores/teensy3
 ---> Using cache
 ---> b46738ff4dc8
Step 28/37 : ADD internal/CMake/select_all/CMakeLists.txt /teensyduino/cores/teensy4
 ---> Using cache
 ---> c5afb38c8bf6
Step 29/37 : ADD internal/CMake/select_all/CMakeLists.txt /teensyduino/src
 ---> Using cache
 ---> 82e7d6cfe0d6
Step 30/37 : RUN rm /teensyduino/cores/teensy3/main.* &&     rm /teensyduino/cores/teensy4/main.* &&     rm /teensyduino/cores/teensy4/Blink.cc
 ---> Using cache
 ---> dc82dbd9a69d
Step 31/37 : WORKDIR /helper_scripts
 ---> Using cache
 ---> 079c9268c11b
Step 32/37 : ADD internal/internal_scripts/* /helper_scripts/
 ---> Using cache
 ---> 3beed0c1eaa3
Step 33/37 : ENV PYTHONPATH="${PYTHONPATH}:/helper_scripts"
 ---> Using cache
 ---> 47ce00628516
Step 34/37 : RUN cp /teensyduino/bin/arm-none-eabi-as /usr/bin/as
 ---> Using cache
 ---> e45a5c976edd
Step 35/37 : WORKDIR /teensyduino
 ---> Using cache
 ---> ba0349f757b5
Step 36/37 : RUN rm -rf ./${DOCKER_ARDUINO_VERSION} &&     rm TeensyduinoInstall.linux64
 ---> Using cache
 ---> 9b7626313339
Step 37/37 : CMD ["/bin/bash","/helper_scripts/entrypoint.sh"]
 ---> Using cache
 ---> c0e4e965732f
Successfully built c0e4e965732f
Successfully tagged teensy_dev:latest
cp: cannot stat '/libs/*': No such file or directory
/teensyduino/libraries
-- Compiling for Teensy 4.1
-- Using Teensy4 Core
-- /teensyduino/libraries
-- AccelStepper/src
-- Adafruit_nRF8001/utility
-- Audio/utility
-- Bounce2/src
-- CryptoAccel/src
-- Encoder/utility
-- Ethernet/src
-- Ethernet/src/utility
-- FNET/src
-- FastLED/src
-- FlexIO_t4/src
-- ILI9488_t3/src
-- IRremote/src
-- Keypad/src
-- LedControl/src
-- LiquidCrystal/src
-- LittleFS/src
-- MFRC522/src
-- MIDI/src
-- NXPMotionSense/utility
-- NativeEthernet/src
-- NativeEthernet/src/utility
-- PS2Keyboard/utility
-- ResponsiveAnalogRead/src
-- SD/src
-- SdFat/src
-- Snooze/src
-- USBHost_t36/utility
-- Wire/utility
-- ADC
-- AccelStepper
-- Adafruit-GFX-Library
-- Adafruit_BusIO
-- Adafruit_NeoPixel
-- Adafruit_STMPE610
-- Adafruit_VS1053
-- Adafruit_nRF8001
-- AltSoftSerial
-- Artnet
-- Audio
-- Bounce
-- Bounce2
-- CapacitiveSensor
-- CryptoAccel
-- DS1307RTC
-- DmxSimple
-- DogLcd
-- EEPROM
-- EasyTransfer
-- EasyTransferI2C
-- Encoder
-- Entropy
-- Ethernet
-- FNET
-- FastCRC
-- FastLED
-- FlexCAN
-- FlexCAN_T4
-- FlexIO_t4
-- FlexiTimer2
-- FreqCount
-- FreqMeasure
-- FreqMeasureMulti
-- FrequencyTimer2
-- ILI9341_t3
-- ILI9488_t3
-- IRremote
-- Keypad
-- LedControl
-- LedDisplay
-- LiquidCrystal
-- LiquidCrystalFast
-- LittleFS
-- LowPower
-- MFRC522
-- MIDI
-- Metro
-- MsTimer2
-- NXPMotionSense
-- NativeEthernet
-- OSC
-- OctoWS2811
-- OneWire
-- PS2Keyboard
-- PWMServo
-- Ping
-- PulsePosition
-- QuadEncoder
-- RA8875
-- RadioHead
-- ResponsiveAnalogRead
-- SD
-- SPI
-- SPIFlash
-- ST7735_t3
-- SdFat
-- SerialFlash
-- Servo
-- ShiftPWM
-- Snooze
-- SoftPWM
-- SoftwareSerial
-- TFT_ILI9163C
-- Talkie
-- TeensyThreads
-- Time
-- TimeAlarms
-- TimerOne
-- TimerThree
-- TinyGPS
-- Tlc5940
-- TouchScreen
-- USBHost_t36
-- UTFT
-- VirtualWire
-- WS2812Serial
-- Wire
-- XBee
-- XPT2046_Touchscreen
-- i2c_t3
-- ks0108
-- ssd1351
-- x10
-- /teensyduino/libraries/AccelStepper/src/teensyduino/libraries/Adafruit_nRF8001/utility/teensyduino/libraries/Audio/utility/teensyduino/libraries/Bounce2/src/teensyduino/libraries/CryptoAccel/src/teensyduino/libraries/Encoder/utility/teensyduino/libraries/Ethernet/src/teensyduino/libraries/Ethernet/src/utility/teensyduino/libraries/FNET/src/teensyduino/libraries/FastLED/src/teensyduino/libraries/FlexIO_t4/src/teensyduino/libraries/ILI9488_t3/src/teensyduino/libraries/IRremote/src/teensyduino/libraries/Keypad/src/teensyduino/libraries/LedControl/src/teensyduino/libraries/LiquidCrystal/src/teensyduino/libraries/LittleFS/src/teensyduino/libraries/MFRC522/src/teensyduino/libraries/MIDI/src/teensyduino/libraries/NXPMotionSense/utility/teensyduino/libraries/NativeEthernet/src/teensyduino/libraries/NativeEthernet/src/utility/teensyduino/libraries/PS2Keyboard/utility/teensyduino/libraries/ResponsiveAnalogRead/src/teensyduino/libraries/SD/src/teensyduino/libraries/SdFat/src/teensyduino/libraries/Snooze/src/teensyduino/libraries/USBHost_t36/utility/teensyduino/libraries/Wire/utilityADCAccelStepperAdafruit-GFX-LibraryAdafruit_BusIOAdafruit_NeoPixelAdafruit_STMPE610ArtnetAudioBounceBounce2CapacitiveSensorCryptoAccelDS1307RTCDmxSimpleDogLcdEEPROMEasyTransferEasyTransferI2CEncoderEntropyEthernetFNETFastCRCFastLEDFlexCAN_T4FlexIO_t4FlexiTimer2FreqCountFreqMeasureFreqMeasureMultiILI9341_t3ILI9488_t3IRremoteKeypadLedControlLedDisplayLiquidCrystalLiquidCrystalFastLittleFSMFRC522MIDIMetroMsTimer2NXPMotionSenseNativeEthernetOSCOctoWS2811OneWirePS2KeyboardPWMServoPingPulsePositionQuadEncoderRA8875RadioHeadResponsiveAnalogReadSDSPISPIFlashST7735_t3SdFatSerialFlashServoShiftPWMSnoozeSoftPWMSoftwareSerialTFT_ILI9163CTalkieTeensyThreadsTimeTimeAlarmsTimerOneTimerThreeTinyGPSTouchScreenUSBHost_t36WS2812SerialWireXBeeXPT2046_Touchscreeni2c_t3ssd1351x10
-- ADCAccelStepperAdafruit-GFX-LibraryAdafruit_BusIOAdafruit_NeoPixelAdafruit_STMPE610ArtnetAudioBounceBounce2CapacitiveSensorCryptoAccelDS1307RTCDmxSimpleDogLcdEEPROMEasyTransferEasyTransferI2CEncoderEntropyEthernetFNETFastCRCFastLEDFlexCAN_T4FlexIO_t4FlexiTimer2FreqCountFreqMeasureFreqMeasureMultiILI9341_t3ILI9488_t3IRremoteKeypadLedControlLedDisplayLiquidCrystalLiquidCrystalFastLittleFSMFRC522MIDIMetroMsTimer2NXPMotionSenseNativeEthernetOSCOctoWS2811OneWirePS2KeyboardPWMServoPingPulsePositionQuadEncoderRA8875RadioHeadResponsiveAnalogReadSDSPISPIFlashST7735_t3SdFatSerialFlashServoShiftPWMSnoozeSoftPWMSoftwareSerialTFT_ILI9163CTalkieTeensyThreadsTimeTimeAlarmsTimerOneTimerThreeTinyGPSTouchScreenUSBHost_t36WS2812SerialWireXBeeXPT2046_Touchscreeni2c_t3ssd1351x10
-- /teensyduino/libraries/AccelStepper/src/teensyduino/libraries/Adafruit_nRF8001/utility/teensyduino/libraries/Audio/utility/teensyduino/libraries/Bounce2/src/teensyduino/libraries/CryptoAccel/src/teensyduino/libraries/Encoder/utility/teensyduino/libraries/Ethernet/src/teensyduino/libraries/Ethernet/src/utility/teensyduino/libraries/FNET/src/teensyduino/libraries/FastLED/src/teensyduino/libraries/FlexIO_t4/src/teensyduino/libraries/ILI9488_t3/src/teensyduino/libraries/IRremote/src/teensyduino/libraries/Keypad/src/teensyduino/libraries/LedControl/src/teensyduino/libraries/LiquidCrystal/src/teensyduino/libraries/LittleFS/src/teensyduino/libraries/MFRC522/src/teensyduino/libraries/MIDI/src/teensyduino/libraries/NXPMotionSense/utility/teensyduino/libraries/NativeEthernet/src/teensyduino/libraries/NativeEthernet/src/utility/teensyduino/libraries/PS2Keyboard/utility/teensyduino/libraries/ResponsiveAnalogRead/src/teensyduino/libraries/SD/src/teensyduino/libraries/SdFat/src/teensyduino/libraries/Snooze/src/teensyduino/libraries/USBHost_t36/utility/teensyduino/libraries/Wire/utilityADCAccelStepperAdafruit-GFX-LibraryAdafruit_BusIOAdafruit_NeoPixelAdafruit_STMPE610ArtnetAudioBounceBounce2CapacitiveSensorCryptoAccelDS1307RTCDmxSimpleDogLcdEEPROMEasyTransferEasyTransferI2CEncoderEntropyEthernetFNETFastCRCFastLEDFlexCAN_T4FlexIO_t4FlexiTimer2FreqCountFreqMeasureFreqMeasureMultiILI9341_t3ILI9488_t3IRremoteKeypadLedControlLedDisplayLiquidCrystalLiquidCrystalFastLittleFSMFRC522MIDIMetroMsTimer2NXPMotionSenseNativeEthernetOSCOctoWS2811OneWirePS2KeyboardPWMServoPingPulsePositionQuadEncoderRA8875RadioHeadResponsiveAnalogReadSDSPISPIFlashST7735_t3SdFatSerialFlashServoShiftPWMSnoozeSoftPWMSoftwareSerialTFT_ILI9163CTalkieTeensyThreadsTimeTimeAlarmsTimerOneTimerThreeTinyGPSTouchScreenUSBHost_t36WS2812SerialWireXBeeXPT2046_Touchscreeni2c_t3ssd1351x10
-- Configuring done
-- Generating done
-- Build files have been written to: /teensyduino/build
Scanning dependencies of target x10
Scanning dependencies of target Adafruit-GFX-Library
Scanning dependencies of target AccelStepper
Scanning dependencies of target ADC
Scanning dependencies of target Adafruit_STMPE610
Scanning dependencies of target Adafruit_BusIO
Scanning dependencies of target DmxSimple
Scanning dependencies of target Artnet
Scanning dependencies of target Adafruit_NeoPixel
Scanning dependencies of target Bounce2
Scanning dependencies of target CapacitiveSensor
Scanning dependencies of target CryptoAccel
Scanning dependencies of target DS1307RTC
Scanning dependencies of target DogLcd
[  0%] Building CXX object libraries/x10/CMakeFiles/x10.dir/x10.cpp.obj
Scanning dependencies of target Bounce
[  0%] Building CXX object libraries/Adafruit-GFX-Library/CMakeFiles/Adafruit-GFX-Library.dir/Adafruit_GFX.cpp.obj
[  0%] Building CXX object libraries/Adafruit_STMPE610/CMakeFiles/Adafruit_STMPE610.dir/Adafruit_STMPE610.cpp.obj
[  0%] Building CXX object libraries/ADC/CMakeFiles/ADC.dir/ADC.cpp.obj
[  0%] Building CXX object libraries/Adafruit_BusIO/CMakeFiles/Adafruit_BusIO.dir/Adafruit_BusIO_Register.cpp.obj
[  1%] Building CXX object libraries/AccelStepper/CMakeFiles/AccelStepper.dir/src/AccelStepper.cpp.obj
[  1%] Building CXX object libraries/CapacitiveSensor/CMakeFiles/CapacitiveSensor.dir/CapacitiveSensor.cpp.obj
[  1%] Building ASM object libraries/CryptoAccel/CMakeFiles/CryptoAccel.dir/src/mmcau_aes_functions.S.obj
[  1%] Building CXX object libraries/Artnet/CMakeFiles/Artnet.dir/Artnet.cpp.obj
[  1%] Building CXX object libraries/Bounce2/CMakeFiles/Bounce2.dir/src/Bounce2.cpp.obj
[  1%] Building CXX object libraries/Adafruit_NeoPixel/CMakeFiles/Adafruit_NeoPixel.dir/Adafruit_NeoPixel.cpp.obj
[  2%] Building CXX object libraries/DS1307RTC/CMakeFiles/DS1307RTC.dir/DS1307RTC.cpp.obj
[  2%] Building CXX object libraries/DmxSimple/CMakeFiles/DmxSimple.dir/DmxSimple.cpp.obj
Scanning dependencies of target Audio
[  2%] Building CXX object libraries/DogLcd/CMakeFiles/DogLcd.dir/DogLcd.cpp.obj
[  2%] Building CXX object libraries/Bounce/CMakeFiles/Bounce.dir/Bounce.cpp.obj
[  2%] Building ASM object libraries/CryptoAccel/CMakeFiles/CryptoAccel.dir/src/mmcau_des_functions.S.obj
[  3%] Building ASM object libraries/CryptoAccel/CMakeFiles/CryptoAccel.dir/src/mmcau_md5_functions.S.obj
[  3%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/Quantizer.cpp.obj
[  4%] Linking CXX static library libx10.a
[  4%] Building ASM object libraries/CryptoAccel/CMakeFiles/CryptoAccel.dir/src/mmcau_sha1_functions.S.obj
[  5%] Building CXX object libraries/Adafruit_BusIO/CMakeFiles/Adafruit_BusIO.dir/Adafruit_I2CDevice.cpp.obj
[  5%] Linking CXX static library libCapacitiveSensor.a
[  6%] Linking CXX static library libArtnet.a
[  7%] Linking CXX static library libBounce2.a
[  7%] Linking CXX static library libBounce.a
[  7%] Built target x10
[  7%] Building ASM object libraries/CryptoAccel/CMakeFiles/CryptoAccel.dir/src/mmcau_sha256_functions.S.obj
[  7%] Building CXX object libraries/ADC/CMakeFiles/ADC.dir/ADC_Module.cpp.obj
[  7%] Building CXX object libraries/AccelStepper/CMakeFiles/AccelStepper.dir/src/MultiStepper.cpp.obj
[  7%] Linking CXX static library libDmxSimple.a
[  8%] Linking CXX static library libDogLcd.a
[  8%] Building C object libraries/Adafruit_NeoPixel/CMakeFiles/Adafruit_NeoPixel.dir/esp8266.c.obj
[  8%] Built target Bounce
[  8%] Built target Artnet
[  8%] Linking CXX static library libDS1307RTC.a
[  8%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/Resampler.cpp.obj
[  8%] Linking CXX static library libCryptoAccel.a
[  8%] Building CXX object libraries/ADC/CMakeFiles/ADC.dir/AnalogBufferDMA.cpp.obj
[  8%] Built target CapacitiveSensor
[  8%] Built target DmxSimple
[  8%] Built target Bounce2
Scanning dependencies of target EEPROM
[  8%] Building CXX object libraries/Adafruit_BusIO/CMakeFiles/Adafruit_BusIO.dir/Adafruit_SPIDevice.cpp.obj
[  9%] Linking CXX static library libAdafruit_NeoPixel.a
Scanning dependencies of target EasyTransferI2C
Scanning dependencies of target EasyTransfer
[  9%] Building CXX object libraries/EEPROM/CMakeFiles/EEPROM.dir/EEPROM.cpp.obj
[  9%] Built target DS1307RTC
[  9%] Built target DogLcd
[  9%] Built target CryptoAccel
[ 10%] Building CXX object libraries/EasyTransferI2C/CMakeFiles/EasyTransferI2C.dir/EasyTransferI2C.cpp.obj
[ 10%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/analyze_fft1024.cpp.obj
[ 10%] Building CXX object libraries/EasyTransfer/CMakeFiles/EasyTransfer.dir/EasyTransfer.cpp.obj
[ 10%] Built target Adafruit_NeoPixel
Scanning dependencies of target Encoder
[ 10%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/analyze_fft256.cpp.obj
[ 10%] Building CXX object libraries/Adafruit-GFX-Library/CMakeFiles/Adafruit-GFX-Library.dir/Adafruit_GrayOLED.cpp.obj
Scanning dependencies of target Entropy
[ 11%] Building CXX object libraries/Adafruit-GFX-Library/CMakeFiles/Adafruit-GFX-Library.dir/Adafruit_SPITFT.cpp.obj
[ 11%] Building CXX object libraries/Encoder/CMakeFiles/Encoder.dir/Encoder.cpp.obj
[ 12%] Building CXX object libraries/Entropy/CMakeFiles/Entropy.dir/Entropy.cpp.obj
[ 12%] Linking CXX static library libAccelStepper.a
[ 12%] Linking CXX static library libAdafruit_STMPE610.a
[ 12%] Building C object libraries/Adafruit-GFX-Library/CMakeFiles/Adafruit-GFX-Library.dir/glcdfont.c.obj
[ 12%] Built target AccelStepper
[ 12%] Built target Adafruit_STMPE610
[ 12%] Linking CXX static library libEEPROM.a
Scanning dependencies of target Ethernet
[ 12%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/analyze_peak.cpp.obj
[ 13%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/analyze_notefreq.cpp.obj
[ 13%] Built target FNET
[ 13%] Building CXX object libraries/Ethernet/CMakeFiles/Ethernet.dir/src/Dhcp.cpp.obj
Scanning dependencies of target FastCRC
[ 13%] Linking CXX static library libEncoder.a
Scanning dependencies of target FastLED
[ 13%] Linking CXX static library libADC.a
[ 13%] Building CXX object libraries/FastCRC/CMakeFiles/FastCRC.dir/FastCRChw.cpp.obj
[ 13%] Linking CXX static library libEasyTransfer.a
[ 13%] Built target EEPROM
[ 13%] Linking CXX static library libEasyTransferI2C.a
[ 13%] Built target Encoder
[ 13%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/FastLED.cpp.obj
[ 13%] Built target FlexCAN_T4
Scanning dependencies of target FlexIO_t4
Scanning dependencies of target FlexiTimer2
[ 13%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/analyze_print.cpp.obj
[ 13%] Built target EasyTransfer
[ 13%] Linking CXX static library libAdafruit_BusIO.a
[ 13%] Building CXX object libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/src/FlexIOSPI.cpp.obj
[ 13%] Built target EasyTransferI2C
[ 13%] Built target ADC
[ 13%] Building CXX object libraries/FlexiTimer2/CMakeFiles/FlexiTimer2.dir/FlexiTimer2.cpp.obj
Scanning dependencies of target FreqCount
Scanning dependencies of target FreqMeasure
[ 14%] Building CXX object libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/src/FlexIO_t4.cpp.obj
[ 14%] Building CXX object libraries/FreqCount/CMakeFiles/FreqCount.dir/FreqCount.cpp.obj
[ 14%] Building CXX object libraries/FreqMeasure/CMakeFiles/FreqMeasure.dir/FreqMeasure.cpp.obj
[ 14%] Building CXX object libraries/FastCRC/CMakeFiles/FastCRC.dir/FastCRCsw.cpp.obj
[ 14%] Built target Adafruit_BusIO
Scanning dependencies of target FreqMeasureMulti
[ 14%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/analyze_rms.cpp.obj
[ 14%] Linking CXX static library libEntropy.a
Scanning dependencies of target ILI9341_t3
[ 15%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/analyze_tonedetect.cpp.obj
[ 16%] Building CXX object libraries/FreqMeasureMulti/CMakeFiles/FreqMeasureMulti.dir/FreqMeasureMulti.cpp.obj
[ 16%] Building CXX object libraries/ILI9341_t3/CMakeFiles/ILI9341_t3.dir/ILI9341_t3.cpp.obj
[ 16%] Built target Entropy
[ 17%] Linking CXX static library libFlexiTimer2.a
[ 17%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/async_input_spdif3.cpp.obj
[ 17%] Linking CXX static library libFreqMeasure.a
Scanning dependencies of target ILI9488_t3
[ 17%] Linking CXX static library libFreqCount.a
[ 17%] Building CXX object libraries/ILI9488_t3/CMakeFiles/ILI9488_t3.dir/ILI9488_t3..cpp.obj
[ 17%] Building CXX object libraries/Ethernet/CMakeFiles/Ethernet.dir/src/Dns.cpp.obj
[ 17%] Built target FlexiTimer2
[ 17%] Built target FreqMeasure
[ 17%] Building CXX object libraries/Ethernet/CMakeFiles/Ethernet.dir/src/Ethernet.cpp.obj
[ 17%] Building CXX object libraries/FreqMeasureMulti/CMakeFiles/FreqMeasureMulti.dir/FreqMeasureMultiIMXRT.cpp.obj
Scanning dependencies of target Keypad
[ 17%] Built target FreqCount
Scanning dependencies of target LedControl
[ 17%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/control_ak4558.cpp.obj
[ 17%] Building CXX object libraries/Keypad/CMakeFiles/Keypad.dir/src/Key.cpp.obj
Scanning dependencies of target LedDisplay
[ 18%] Built target IRremote
[ 18%] Building CXX object libraries/ILI9488_t3/CMakeFiles/ILI9488_t3.dir/src/ILI9488_t3.cpp.obj
[ 18%] Building CXX object libraries/LedControl/CMakeFiles/LedControl.dir/src/LedControl.cpp.obj
[ 18%] Building CXX object libraries/LedDisplay/CMakeFiles/LedDisplay.dir/LedDisplay.cpp.obj
Scanning dependencies of target LiquidCrystal
[ 19%] Building CXX object libraries/LiquidCrystal/CMakeFiles/LiquidCrystal.dir/LiquidCrystal.cpp.obj
[ 19%] Building CXX object libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/src/FlexSerial.cpp.obj
[ 19%] Linking CXX static library libAdafruit-GFX-Library.a
[ 19%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/bitswap.cpp.obj
[ 19%] Building CXX object libraries/LiquidCrystal/CMakeFiles/LiquidCrystal.dir/src/LiquidCrystal.cpp.obj
Scanning dependencies of target LiquidCrystalFast
[ 19%] Built target Adafruit-GFX-Library
[ 19%] Building CXX object libraries/Keypad/CMakeFiles/Keypad.dir/src/Keypad.cpp.obj
[ 19%] Building CXX object libraries/LiquidCrystalFast/CMakeFiles/LiquidCrystalFast.dir/LiquidCrystalFast.cpp.obj
[ 19%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/control_cs42448.cpp.obj
[ 20%] Building CXX object libraries/Ethernet/CMakeFiles/Ethernet.dir/src/EthernetClient.cpp.obj
[ 20%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/control_cs4272.cpp.obj
[ 21%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/control_sgtl5000.cpp.obj
[ 22%] Linking CXX static library libFastCRC.a
[ 22%] Linking CXX static library libFreqMeasureMulti.a
[ 22%] Linking CXX static library libFlexIO_t4.a
[ 22%] Linking CXX static library libLedControl.a
[ 22%] Built target FastCRC
Scanning dependencies of target LittleFS
[ 22%] Linking CXX static library libLedDisplay.a
[ 22%] Building CXX object libraries/LittleFS/CMakeFiles/LittleFS.dir/src/LittleFS.cpp.obj
[ 22%] Built target FreqMeasureMulti
[ 22%] Built target FlexIO_t4
Scanning dependencies of target MFRC522
[ 22%] Building CXX object libraries/LittleFS/CMakeFiles/LittleFS.dir/src/LittleFS_NAND.cpp.obj
[ 22%] Linking CXX static library libLiquidCrystal.a
[ 23%] Linking CXX static library libKeypad.a
[ 23%] Building CXX object libraries/MFRC522/CMakeFiles/MFRC522.dir/src/MFRC522.cpp.obj
[ 23%] Built target LedControl
[ 23%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/colorpalettes.cpp.obj
Scanning dependencies of target MIDI
[ 24%] Linking CXX static library libLiquidCrystalFast.a
[ 24%] Built target LedDisplay
[ 24%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/control_tlv320aic3206.cpp.obj
[ 25%] Building CXX object libraries/MFRC522/CMakeFiles/MFRC522.dir/src/MFRC522Extended.cpp.obj
[ 25%] Building CXX object libraries/MFRC522/CMakeFiles/MFRC522.dir/src/MFRC522Debug.cpp.obj
[ 25%] Building CXX object libraries/MIDI/CMakeFiles/MIDI.dir/src/MIDI.cpp.obj
[ 25%] Built target LiquidCrystal
[ 25%] Built target Keypad
[ 26%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/colorutils.cpp.obj
Scanning dependencies of target Metro
[ 26%] Building CXX object libraries/Ethernet/CMakeFiles/Ethernet.dir/src/EthernetServer.cpp.obj
[ 27%] Building CXX object libraries/Metro/CMakeFiles/Metro.dir/Metro.cpp.obj
[ 27%] Built target LiquidCrystalFast
Scanning dependencies of target MsTimer2
[ 27%] Building CXX object libraries/Ethernet/CMakeFiles/Ethernet.dir/src/EthernetUdp.cpp.obj
[ 27%] Building CXX object libraries/MsTimer2/CMakeFiles/MsTimer2.dir/MsTimer2.cpp.obj
[ 27%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/control_wm8731.cpp.obj
[ 27%] Linking CXX static library libMetro.a
[ 27%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/hsv2rgb.cpp.obj
[ 27%] Linking CXX static library libMIDI.a
[ 27%] Built target Metro
[ 27%] Building CXX object libraries/MFRC522/CMakeFiles/MFRC522.dir/src/MFRC522Hack.cpp.obj
[ 27%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/lib8tion.cpp.obj
[ 27%] Built target MIDI
[ 27%] Building C object libraries/Audio/CMakeFiles/Audio.dir/data_bandlimit_step.c.obj
[ 28%] Building C object libraries/ILI9341_t3/CMakeFiles/ILI9341_t3.dir/font_Arial.c.obj
[ 28%] Building C object libraries/Audio/CMakeFiles/Audio.dir/data_spdif.c.obj
[ 28%] Building CXX object libraries/Ethernet/CMakeFiles/Ethernet.dir/src/socket.cpp.obj
[ 29%] Linking CXX static library libMsTimer2.a
[ 29%] Building CXX object libraries/Ethernet/CMakeFiles/Ethernet.dir/src/utility/w5100.cpp.obj
[ 30%] Building C object libraries/Audio/CMakeFiles/Audio.dir/data_ulaw.c.obj
[ 30%] Building C object libraries/Audio/CMakeFiles/Audio.dir/data_waveforms.c.obj
[ 30%] Building C object libraries/Audio/CMakeFiles/Audio.dir/data_windows.c.obj
[ 30%] Built target MsTimer2
[ 30%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_bitcrusher.cpp.obj
[ 31%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_chorus.cpp.obj
[ 31%] Building C object libraries/ILI9488_t3/CMakeFiles/ILI9488_t3.dir/src/glcdfont.c.obj
Scanning dependencies of target NXPMotionSense
[ 31%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/noise.cpp.obj
[ 32%] Building CXX object libraries/NXPMotionSense/CMakeFiles/NXPMotionSense.dir/NXPMotionSense.cpp.obj
[ 32%] Building C object libraries/ILI9341_t3/CMakeFiles/ILI9341_t3.dir/font_ArialBold.c.obj
[ 32%] Building C object libraries/ILI9488_t3/CMakeFiles/ILI9488_t3.dir/src/ili9488_t3_font_Arial.c.obj
[ 32%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/platforms.cpp.obj
[ 34%] Built target NativeEthernet
[ 34%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_combine.cpp.obj
[ 34%] Building CXX object libraries/NXPMotionSense/CMakeFiles/NXPMotionSense.dir/SensorFusion.cpp.obj
[ 35%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/power_mgt.cpp.obj
[ 35%] Linking CXX static library libMFRC522.a
[ 35%] Building CXX object libraries/FastLED/CMakeFiles/FastLED.dir/src/wiring.cpp.obj
[ 35%] Building C object libraries/NXPMotionSense/CMakeFiles/NXPMotionSense.dir/matrix.c.obj
[ 35%] Built target MFRC522
Scanning dependencies of target OSC
[ 35%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_delay.cpp.obj
[ 35%] Building C object libraries/ILI9488_t3/CMakeFiles/ILI9488_t3.dir/src/ili9488_t3_font_ArialBold.c.obj
[ 35%] Building CXX object libraries/OSC/CMakeFiles/OSC.dir/OSCBundle.cpp.obj
[ 35%] Building CXX object libraries/OSC/CMakeFiles/OSC.dir/OSCData.cpp.obj
[ 35%] Building C object libraries/ILI9341_t3/CMakeFiles/ILI9341_t3.dir/glcdfont.c.obj
[ 35%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_delay_ext.cpp.obj
Scanning dependencies of target OctoWS2811
[ 35%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_envelope.cpp.obj
[ 36%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_fade.cpp.obj
[ 36%] Linking CXX static library libLittleFS.a
[ 36%] Building CXX object libraries/OctoWS2811/CMakeFiles/OctoWS2811.dir/OctoWS2811.cpp.obj
[ 36%] Built target LittleFS
[ 37%] Building C object libraries/OSC/CMakeFiles/OSC.dir/OSCMatch.c.obj
[ 37%] Building CXX object libraries/OSC/CMakeFiles/OSC.dir/OSCMessage.cpp.obj
[ 37%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_flange.cpp.obj
[ 38%] Building C object libraries/ILI9488_t3/CMakeFiles/ILI9488_t3.dir/src/ili9488_t3_font_ComicSansMS.c.obj
[ 38%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_freeverb.cpp.obj
[ 38%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_granular.cpp.obj
[ 38%] Building CXX object libraries/OSC/CMakeFiles/OSC.dir/OSCTiming.cpp.obj
[ 38%] Building CXX object libraries/OSC/CMakeFiles/OSC.dir/SLIPEncodedSerial.cpp.obj
[ 38%] Building CXX object libraries/OctoWS2811/CMakeFiles/OctoWS2811.dir/OctoWS2811_imxrt.cpp.obj
[ 39%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_midside.cpp.obj
[ 40%] Building CXX object libraries/OSC/CMakeFiles/OSC.dir/SLIPEncodedUSBSerial.cpp.obj
[ 40%] Linking CXX static library libNXPMotionSense.a
[ 40%] Linking CXX static library libEthernet.a
[ 40%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_multiply.cpp.obj
Scanning dependencies of target OneWire
[ 40%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_rectifier.cpp.obj
[ 40%] Built target NXPMotionSense
[ 41%] Building CXX object libraries/OneWire/CMakeFiles/OneWire.dir/OneWire.cpp.obj
[ 41%] Built target Ethernet
[ 41%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_reverb.cpp.obj
Scanning dependencies of target PS2Keyboard
Scanning dependencies of target PWMServo
[ 41%] Building CXX object libraries/PWMServo/CMakeFiles/PWMServo.dir/PWMServo.cpp.obj
[ 41%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_wavefolder.cpp.obj
[ 42%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/effect_waveshaper.cpp.obj
[ 42%] Building CXX object libraries/PS2Keyboard/CMakeFiles/PS2Keyboard.dir/PS2Keyboard.cpp.obj
[ 42%] Linking CXX static library libFastLED.a
[ 42%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/filter_biquad.cpp.obj
[ 42%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/filter_fir.cpp.obj
[ 42%] Built target FastLED
[ 42%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/filter_ladder.cpp.obj
[ 43%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/filter_variable.cpp.obj
Scanning dependencies of target Ping
[ 43%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_adc.cpp.obj
[ 43%] Building CXX object libraries/Ping/CMakeFiles/Ping.dir/Ping.cpp.obj
[ 43%] Linking CXX static library libOctoWS2811.a
[ 44%] Linking CXX static library libPWMServo.a
[ 44%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_adcs.cpp.obj
[ 44%] Linking CXX static library libPS2Keyboard.a
[ 44%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_i2s.cpp.obj
[ 44%] Linking CXX static library libOSC.a
[ 44%] Built target PWMServo
[ 44%] Linking CXX static library libOneWire.a
Scanning dependencies of target PulsePosition
[ 44%] Built target OctoWS2811
[ 44%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_i2s2.cpp.obj
[ 45%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_i2s_hex.cpp.obj
[ 45%] Building CXX object libraries/PulsePosition/CMakeFiles/PulsePosition.dir/PulsePosition.cpp.obj
[ 45%] Built target OSC
[ 45%] Built target OneWire
[ 45%] Built target QuadEncoder
[ 45%] Built target PS2Keyboard
Scanning dependencies of target RA8875
Scanning dependencies of target ResponsiveAnalogRead
Scanning dependencies of target RadioHead
Scanning dependencies of target SD
[ 45%] Building CXX object libraries/RA8875/CMakeFiles/RA8875.dir/RA8875.cpp.obj
[ 45%] Linking CXX static library libPing.a
[ 45%] Building CXX object libraries/ResponsiveAnalogRead/CMakeFiles/ResponsiveAnalogRead.dir/src/ResponsiveAnalogRead.cpp.obj
[ 46%] Building CXX object libraries/PulsePosition/CMakeFiles/PulsePosition.dir/PulsePositionIMXRT.cpp.obj
[ 46%] Building CXX object libraries/SD/CMakeFiles/SD.dir/src/SD.cpp.obj
[ 46%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_i2s_oct.cpp.obj
[ 46%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHCRC.cpp.obj
[ 46%] Built target Ping
Scanning dependencies of target SPIFlash
Scanning dependencies of target ST7735_t3
[ 46%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_i2s_quad.cpp.obj
Scanning dependencies of target SdFat
Scanning dependencies of target SPI
[ 47%] Building CXX object libraries/SdFat/CMakeFiles/SdFat.dir/src/FreeStack.cpp.obj
[ 47%] Building CXX object libraries/SPIFlash/CMakeFiles/SPIFlash.dir/SPIFlash.cpp.obj
[ 47%] Building CXX object libraries/SPI/CMakeFiles/SPI.dir/SPI.cpp.obj
[ 47%] Building CXX object libraries/ST7735_t3/CMakeFiles/ST7735_t3.dir/ST7735_t3.cpp.obj
Scanning dependencies of target SerialFlash
[ 47%] Building CXX object libraries/SerialFlash/CMakeFiles/SerialFlash.dir/SerialFlashChip.cpp.obj
[ 47%] Building CXX object libraries/ST7735_t3/CMakeFiles/ST7735_t3.dir/ST7789_t3.cpp.obj
[ 47%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_pdm.cpp.obj
[ 47%] Building CXX object libraries/SdFat/CMakeFiles/SdFat.dir/src/MinimumSerial.cpp.obj
[ 48%] Linking CXX static library libResponsiveAnalogRead.a
[ 49%] Building C object libraries/ST7735_t3/CMakeFiles/ST7735_t3.dir/glcdfont.c.obj
[ 49%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHDatagram.cpp.obj
[ 49%] Building C object libraries/ST7735_t3/CMakeFiles/ST7735_t3.dir/st7735_t3_font_Arial.c.obj
[ 49%] Built target ResponsiveAnalogRead
[ 49%] Building CXX object libraries/SerialFlash/CMakeFiles/SerialFlash.dir/SerialFlashDirectory.cpp.obj
Scanning dependencies of target Servo
[ 49%] Building CXX object libraries/Servo/CMakeFiles/Servo.dir/Servo.cpp.obj
[ 49%] Linking CXX static library libSdFat.a
[ 49%] Linking CXX static library libPulsePosition.a
[ 49%] Built target SdFat
[ 49%] Built target PulsePosition
Scanning dependencies of target ShiftPWM
[ 50%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHEncryptedDriver.cpp.obj
[ 50%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHGenericDriver.cpp.obj
[ 50%] Building CXX object libraries/ShiftPWM/CMakeFiles/ShiftPWM.dir/CShiftPWM.cpp.obj
Scanning dependencies of target Snooze
[ 51%] Building CXX object libraries/Snooze/CMakeFiles/Snooze.dir/src/Snooze.cpp.obj
[ 51%] Building C object libraries/ST7735_t3/CMakeFiles/ST7735_t3.dir/st7735_t3_font_ComicSansMS.c.obj
[ 51%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_pdm_i2s2.cpp.obj
[ 51%] Linking CXX static library libSPIFlash.a
[ 51%] Linking CXX static library libServo.a
[ 52%] Linking CXX static library libSPI.a
[ 52%] Built target SPIFlash
[ 52%] Building CXX object libraries/Snooze/CMakeFiles/Snooze.dir/src/SnoozeBlock.cpp.obj
[ 52%] Built target Servo
Scanning dependencies of target SoftPWM
Scanning dependencies of target SoftwareSerial
[ 52%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHGenericSPI.cpp.obj
[ 52%] Linking CXX static library libSD.a
[ 52%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHHardwareSP12.cpp.obj
[ 52%] Built target SPI
[ 52%] Building CXX object libraries/SoftwareSerial/CMakeFiles/SoftwareSerial.dir/SoftwareSerial.cpp.obj
[ 53%] Building CXX object libraries/SoftPWM/CMakeFiles/SoftPWM.dir/SoftPWM.cpp.obj
[ 53%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_spdif3.cpp.obj
[ 53%] Built target SD
Scanning dependencies of target TFT_ILI9163C
[ 53%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_tdm.cpp.obj
[ 53%] Building CXX object libraries/TFT_ILI9163C/CMakeFiles/TFT_ILI9163C.dir/TFT_ILI9163C.cpp.obj
[ 53%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/input_tdm2.cpp.obj
[ 53%] Linking CXX static library libSnooze.a
Scanning dependencies of target Talkie
[ 54%] Linking CXX static library libSerialFlash.a
[ 54%] Linking CXX static library libILI9341_t3.a
Scanning dependencies of target TeensyThreads
[ 55%] Building CXX object libraries/Talkie/CMakeFiles/Talkie.dir/Talkie.cpp.obj
[ 55%] Building ASM object libraries/TeensyThreads/CMakeFiles/TeensyThreads.dir/TeensyThreads-asm.S.obj
[ 55%] Built target Snooze
[ 55%] Building CXX object libraries/TeensyThreads/CMakeFiles/TeensyThreads.dir/TeensyThreads.cpp.obj
[ 55%] Linking CXX static library libSoftwareSerial.a
[ 55%] Building ASM object libraries/Audio/CMakeFiles/Audio.dir/memcpy_audio.S.obj
[ 55%] Linking CXX static library libShiftPWM.a
[ 55%] Built target SerialFlash
Scanning dependencies of target Time
[ 55%] Built target ILI9341_t3
[ 56%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/mixer.cpp.obj
[ 56%] Built target SoftwareSerial
Scanning dependencies of target TimeAlarms
[ 56%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHHardwareSP1I.cpp.obj
[ 57%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHHardwareSPI.cpp.obj
[ 57%] Building CXX object libraries/Time/CMakeFiles/Time.dir/DateStrings.cpp.obj
[ 57%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_adat.cpp.obj
[ 57%] Building CXX object libraries/TimeAlarms/CMakeFiles/TimeAlarms.dir/TimeAlarms.cpp.obj
[ 58%] Linking CXX static library libSoftPWM.a
[ 58%] Built target ShiftPWM
Scanning dependencies of target TimerOne
[ 58%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_dac.cpp.obj
[ 58%] Building CXX object libraries/Time/CMakeFiles/Time.dir/Time.cpp.obj
[ 58%] Building CXX object libraries/TimerOne/CMakeFiles/TimerOne.dir/TimerOne.cpp.obj
[ 58%] Built target SoftPWM
[ 58%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_dacs.cpp.obj
[ 59%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_i2s2.cpp.obj
[ 59%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_i2s.cpp.obj
[ 59%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_i2s_hex.cpp.obj
[ 59%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHMesh.cpp.obj
[ 59%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHNRFSPIDriver.cpp.obj
[ 59%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHReliableDatagram.cpp.obj
/teensyduino/libraries/TeensyThreads/TeensyThreads.cpp:371:13: warning: 'void context_pit_empty()' defined but not used [-Wunused-function]
 static void context_pit_empty() {}
             ^
[ 59%] Linking CXX static library libTalkie.a
[ 59%] Linking CXX static library libTFT_ILI9163C.a
[ 59%] Linking CXX static library libTimerOne.a
[ 60%] Linking CXX static library libTeensyThreads.a
[ 61%] Linking CXX static library libTimeAlarms.a
[ 61%] Built target Talkie
[ 61%] Built target TimerOne
[ 61%] Linking CXX static library libILI9488_t3.a
[ 61%] Built target TFT_ILI9163C
[ 61%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_i2s_oct.cpp.obj
[ 62%] Built target TimeAlarms
[ 62%] Building C object libraries/RA8875/CMakeFiles/RA8875.dir/glcdfont.c.obj
Scanning dependencies of target TimerThree
Scanning dependencies of target TouchScreen
[ 62%] Built target TeensyThreads
Scanning dependencies of target TinyGPS
[ 62%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_i2s_quad.cpp.obj
[ 62%] Building CXX object libraries/TimerThree/CMakeFiles/TimerThree.dir/TimerThree.cpp.obj
[ 62%] Building CXX object libraries/TouchScreen/CMakeFiles/TouchScreen.dir/TouchScreen.cpp.obj
[ 63%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHRouter.cpp.obj
[ 63%] Building CXX object libraries/TinyGPS/CMakeFiles/TinyGPS.dir/TinyGPS.cpp.obj
[ 63%] Built target ILI9488_t3
[ 63%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_pt8211.cpp.obj
[ 64%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_mqs.cpp.obj
[ 64%] Linking CXX static library libTime.a
[ 64%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHSPIDriver.cpp.obj
[ 64%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RHSoftwareSPI.cpp.obj
[ 64%] Built target Time
[ 64%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_pt8211_2.cpp.obj
[ 69%] Built target USBHost_t36
Scanning dependencies of target WS2812Serial
[ 69%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_ABZ.cpp.obj
Scanning dependencies of target Wire
[ 69%] Building CXX object libraries/WS2812Serial/CMakeFiles/WS2812Serial.dir/WS2812Serial.cpp.obj
[ 70%] Linking CXX static library libTimerThree.a
[ 70%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_ASK.cpp.obj
[ 70%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_pwm.cpp.obj
[ 70%] Building CXX object libraries/Wire/CMakeFiles/Wire.dir/Wire.cpp.obj
[ 70%] Linking CXX static library libTouchScreen.a
[ 70%] Built target TimerThree
[ 70%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_spdif.cpp.obj
[ 71%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_spdif2.cpp.obj
Scanning dependencies of target XBee
[ 71%] Built target TouchScreen
Scanning dependencies of target XPT2046_Touchscreen
[ 72%] Building CXX object libraries/XBee/CMakeFiles/XBee.dir/XBee.cpp.obj
[ 72%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_spdif3.cpp.obj
[ 72%] Building CXX object libraries/XPT2046_Touchscreen/CMakeFiles/XPT2046_Touchscreen.dir/XPT2046_Touchscreen.cpp.obj
[ 73%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_CC110.cpp.obj
[ 73%] Linking CXX static library libTinyGPS.a
[ 74%] Building CXX object libraries/Wire/CMakeFiles/Wire.dir/WireIMXRT.cpp.obj
[ 74%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_E32.cpp.obj
[ 74%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_tdm.cpp.obj
[ 74%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_MRF89.cpp.obj
[ 74%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/output_tdm2.cpp.obj
[ 74%] Built target TinyGPS
Scanning dependencies of target i2c_t3
[ 74%] Linking CXX static library libWS2812Serial.a
[ 74%] Building CXX object libraries/i2c_t3/CMakeFiles/i2c_t3.dir/i2c_t3.cpp.obj
[ 75%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/play_memory.cpp.obj
[ 75%] Built target WS2812Serial
Scanning dependencies of target ssd1351
[ 76%] Linking CXX static library libi2c_t3.a
[ 76%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_NRF24.cpp.obj
[ 76%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/play_queue.cpp.obj
[ 76%] Building C object libraries/ssd1351/CMakeFiles/ssd1351.dir/glcdfont.c.obj
[ 76%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/play_sd_raw.cpp.obj
[ 76%] Built target i2c_t3
[ 76%] Linking CXX static library libssd1351.a
[ 76%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/play_sd_wav.cpp.obj
[ 76%] Linking CXX static library libXPT2046_Touchscreen.a
[ 76%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/play_serialflash_raw.cpp.obj
[ 76%] Built target ssd1351
[ 76%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_NRF51.cpp.obj
[ 77%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/record_queue.cpp.obj
[ 77%] Building CXX object libraries/Wire/CMakeFiles/Wire.dir/WireKinetis.cpp.obj
[ 78%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_NRF905.cpp.obj
[ 78%] Built target XPT2046_Touchscreen
[ 78%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_RF22.cpp.obj
[ 78%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/spi_interrupt.cpp.obj
[ 78%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_dc.cpp.obj
[ 78%] Building C object libraries/Wire/CMakeFiles/Wire.dir/utility/twi.c.obj
[ 78%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_karplusstrong.cpp.obj
[ 78%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_RF24.cpp.obj
[ 78%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_RF69.cpp.obj
[ 79%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_RF95.cpp.obj
[ 79%] Linking CXX static library libWire.a
[ 79%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_pinknoise.cpp.obj
[ 80%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_pwm.cpp.obj
[ 80%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_simple_drum.cpp.obj
[ 80%] Built target Wire
[ 80%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_Serial.cpp.obj
[ 80%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_sine.cpp.obj
[ 80%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_tonesweep.cpp.obj
[ 80%] Building CXX object libraries/RadioHead/CMakeFiles/RadioHead.dir/RH_TCP.cpp.obj
[ 80%] Linking CXX static library libXBee.a
[ 81%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_waveform.cpp.obj
[ 81%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_wavetable.cpp.obj
[ 81%] Built target XBee
[ 81%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/synth_whitenoise.cpp.obj
[ 81%] Building CXX object libraries/Audio/CMakeFiles/Audio.dir/utility/imxrt_hw.cpp.obj
[ 81%] Building C object libraries/Audio/CMakeFiles/Audio.dir/utility/sqrt_integer.c.obj
[ 81%] Linking CXX static library libRA8875.a
[ 81%] Built target RA8875
[ 81%] Linking CXX static library libRadioHead.a
[ 81%] Built target RadioHead
[ 82%] Linking CXX static library libAudio.a
[ 82%] Built target Audio
[ 82%] Linking CXX static library libST7735_t3.a
[ 82%] Built target ST7735_t3
Scanning dependencies of target MAIN
[ 83%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/AudioStream.cpp.obj
[ 83%] Building CXX object CMakeFiles/MAIN.dir/src/blink.cpp.obj
[ 83%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial1.cpp.obj
[ 83%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/CrashReport.cpp.obj
[ 83%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial6.cpp.obj
[ 83%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial3.cpp.obj
[ 83%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/DMAChannel.cpp.obj
[ 84%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial.cpp.obj
[ 84%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial2.cpp.obj
[ 84%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial4.cpp.obj
[ 84%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial7.cpp.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/IPAddress.cpp.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial5.cpp.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/EventResponder.cpp.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/HardwareSerial8.cpp.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/IntervalTimer.cpp.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/Time.cpp.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/Print.cpp.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/Stream.cpp.obj
[ 86%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/analog.c.obj
[ 86%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/WMath.cpp.obj
[ 87%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/Tone.cpp.obj
[ 87%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/bootdata.c.obj
[ 87%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/WString.cpp.obj
[ 87%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/debugprintf.c.obj
[ 87%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/delay.c.obj
[ 87%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/digital.c.obj
[ 88%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/eeprom.c.obj
[ 89%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/clockspeed.c.obj
[ 89%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/extmem.c.obj
[ 89%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/interrupt.c.obj
[ 89%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/fuse.c.obj
[ 89%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/keylayouts.c.obj
[ 90%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/new.cpp.obj
[ 90%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/pwm.c.obj
[ 90%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/nonstd.c.obj
[ 90%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/rtc.c.obj
[ 91%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent.cpp.obj
[ 91%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent1.cpp.obj
[ 91%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent2.cpp.obj
[ 91%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent3.cpp.obj
[ 92%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent5.cpp.obj
[ 92%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent6.cpp.obj
[ 92%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent4.cpp.obj
[ 92%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent7.cpp.obj
[ 92%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEvent8.cpp.obj
[ 93%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEventUSB1.cpp.obj
[ 93%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/serialEventUSB2.cpp.obj
[ 93%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_alloc_valid.c.obj
[ 93%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_calloc.c.obj
[ 93%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_free.c.obj
[ 94%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_hash.c.obj
[ 94%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_malloc_stats.c.obj
[ 94%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_malloc.c.obj
[ 94%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_pool.c.obj
[ 94%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_realloc_move.c.obj
[ 94%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_realloc.c.obj
[ 95%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_realloc_i.c.obj
[ 95%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_szalloc.c.obj
[ 95%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_util.c.obj
[ 95%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/tempmon.c.obj
[ 96%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/sm_zalloc.c.obj
[ 96%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/startup.c.obj
[ 96%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb.c.obj
[ 97%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_desc.c.obj
[ 97%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/usb_audio.cpp.obj
[ 97%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/usb_flightsim.cpp.obj
[ 98%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_keyboard.c.obj
[ 98%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_joystick.c.obj
[ 98%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/usb_inst.cpp.obj
[ 98%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_mtp.c.obj
[ 98%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_mouse.c.obj
[ 98%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_midi.c.obj
[ 98%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_rawhid.c.obj
[ 99%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_seremu.c.obj
[ 99%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_serial.c.obj
[ 99%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_serial2.c.obj
[ 99%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_serial3.c.obj
[100%] Building C object CMakeFiles/MAIN.dir/cores/teensy4/usb_touch.c.obj
[100%] Building CXX object CMakeFiles/MAIN.dir/cores/teensy4/yield.cpp.obj
[100%] Linking CXX executable MAIN
CMakeFiles/MAIN.dir/cores/teensy4/startup.c.obj: In function `ResetHandler':
/teensyduino/cores/teensy4/startup.c:158: undefined reference to `main'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/MAIN.dir/build.make:1353: MAIN] Error 1
make[1]: *** [CMakeFiles/Makefile2:2192: CMakeFiles/MAIN.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

Attempting to program a Teensy 3.2 using the same sketch, but encountering errors with `FlexIO` 
```
$ ./program.sh 32
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

Sending build context to Docker daemon  158.7kB
Step 1/37 : FROM ubuntu:20.04
 ---> 83a4bf3bb050
Step 2/37 : ARG WORKDIR=/root
 ---> Using cache
 ---> 2088ae0d6f03
Step 3/37 : ENV DOCKER_ARDUINO_VERSION=arduino-1.8.19
 ---> Using cache
 ---> a64f692fe0ef
Step 4/37 : ENV DOCKER_TEENSY_TD_VERSION=td_156
 ---> Using cache
 ---> 47a5dbeecace
Step 5/37 : ENV DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 0db6bea9dd2f
Step 6/37 : RUN apt -y update &&     apt -y install     avr-libc     binutils-avr     cmake     g++     gcc     gcc-avr     git     libfontconfig1     libusb-dev     libxft-dev     make     python3.8     unzip     vim     wget     xz-utils     && apt clean && rm -rf /var/lib/apt/lists
 ---> Using cache
 ---> 52d95ce3b8ee
Step 7/37 : RUN mkdir -p /etc/udev/rules.d/  &&     echo    ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04*", ENV{ID_MM_DEVICE_IGNORE}="1", ENV{ID_MM_PORT_IGNORE}="1"             ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04[789a]*", ENV{MTP_NO_PROBE}="1"             KERNEL=="ttyACM*", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04*", MODE:="0666", RUN:="/bin/stty -F /dev/%k raw -echo"             KERNEL=="hidraw*", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04*", MODE:="0666"             SUBSYSTEMS=="usb", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04*", MODE:="0666" >> /etc/udev/rules.d/00-teensy.rules
 ---> Using cache
 ---> 55655d018e9e
Step 8/37 : WORKDIR /teensyduino
 ---> Using cache
 ---> 1e1ec2ab3bf3
Step 9/37 : RUN wget -q https://downloads.arduino.cc/${DOCKER_ARDUINO_VERSION}-linux64.tar.xz &&     wget -q https://www.pjrc.com/teensy/${DOCKER_TEENSY_TD_VERSION}/TeensyduinoInstall.linux64
 ---> Using cache
 ---> a037ebb3c8ca
Step 10/37 : WORKDIR /teensy_cli
 ---> Using cache
 ---> 0f64ede59e05
Step 11/37 : RUN git clone -q https://github.com/PaulStoffregen/teensy_loader_cli &&     cd teensy_loader_cli &&     make &&     cd rebootor &&     make
 ---> Using cache
 ---> ea238b6a3f9f
Step 12/37 : ENV PATH="$PATH:/teensy_cli/teensy_loader_cli/"
 ---> Using cache
 ---> 5d955a355ef1
Step 13/37 : WORKDIR /teensyduino
 ---> Using cache
 ---> 63b8ca0560ad
Step 14/37 : RUN tar -xf ${DOCKER_ARDUINO_VERSION}-linux64.tar.xz &&     rm ${DOCKER_ARDUINO_VERSION}-linux64.tar.xz &&     chmod 755 TeensyduinoInstall.linux64 &&     ./TeensyduinoInstall.linux64 --dir=/teensyduino/${DOCKER_ARDUINO_VERSION} &&     mkdir -p ./libraries &&     mkdir -p ./cores &&     mkdir -p ./src &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/teensy/avr/libraries/* ./libraries &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/teensy/avr/cores/* ./cores
 ---> Using cache
 ---> e2cdfac9a02d
Step 15/37 : WORKDIR /teensyduino/libraries
 ---> Using cache
 ---> d9b3645a640b
Step 16/37 : RUN      git clone https://github.com/adafruit/Adafruit_BusIO.git &&     git clone https://github.com/adafruit/Adafruit-GFX-Library.git &&     mkdir /src
 ---> Using cache
 ---> 2db85b2ba032
Step 17/37 : WORKDIR /teensyduino/bin
 ---> Using cache
 ---> 787d75edad7b
Step 18/37 : RUN cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/bin/* . &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/lib/gcc/arm-none-eabi/5.4.1/* . &&     cp -u /teensyduino/cores/teensy3/*.ld . &&     cp -u /teensyduino/cores/teensy4/*.ld . &&     mkdir -p /teensyduino/lib/gcc &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/lib/gcc/arm-none-eabi/5.4.1/* /teensyduino/lib/gcc
 ---> Using cache
 ---> 85b0a4e6fb1a
Step 19/37 : WORKDIR /teensyduino/include
 ---> Using cache
 ---> bb851de7420a
Step 20/37 : RUN cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/arm-none-eabi/include/* . &&     cp -r /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/arm-none-eabi/lib/armv7e-m/* . &&     cp /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/arm-none-eabi/lib/nano.specs . &&     cp /teensyduino/${DOCKER_ARDUINO_VERSION}/hardware/tools/arm/arm-none-eabi/lib/lib* . &&     mv crt0.o /teensyduino/lib/gcc
 ---> Using cache
 ---> e2a249f44e74
Step 21/37 : ENV PATH="/teensyduino/bin:/teensyduino/include:/teensyduino/bin/plugin/include:$PATH"
 ---> Using cache
 ---> bd7e94a29c0d
Step 22/37 : WORKDIR /teensyduino
 ---> Using cache
 ---> 5ecfee83f58d
Step 23/37 : RUN mkdir -p /teensyduino/install &&     mkdir -p /teensyduino/build
 ---> Using cache
 ---> 199d9015efdb
Step 24/37 : ADD internal/CMake/project_root/* /teensyduino/
 ---> Using cache
 ---> 96a57088bf92
Step 25/37 : ADD internal/CMake/toolchain/* /teensyduino/
 ---> Using cache
 ---> 10a49ba9c867
Step 26/37 : ADD config.json /teensyduino/
 ---> Using cache
 ---> 539ab76bc6ae
Step 27/37 : ADD internal/CMake/select_all/CMakeLists.txt /teensyduino/cores/teensy3
 ---> Using cache
 ---> b46738ff4dc8
Step 28/37 : ADD internal/CMake/select_all/CMakeLists.txt /teensyduino/cores/teensy4
 ---> Using cache
 ---> c5afb38c8bf6
Step 29/37 : ADD internal/CMake/select_all/CMakeLists.txt /teensyduino/src
 ---> Using cache
 ---> 82e7d6cfe0d6
Step 30/37 : RUN rm /teensyduino/cores/teensy3/main.* &&     rm /teensyduino/cores/teensy4/main.* &&     rm /teensyduino/cores/teensy4/Blink.cc
 ---> Using cache
 ---> dc82dbd9a69d
Step 31/37 : WORKDIR /helper_scripts
 ---> Using cache
 ---> 079c9268c11b
Step 32/37 : ADD internal/internal_scripts/* /helper_scripts/
 ---> Using cache
 ---> 3beed0c1eaa3
Step 33/37 : ENV PYTHONPATH="${PYTHONPATH}:/helper_scripts"
 ---> Using cache
 ---> 47ce00628516
Step 34/37 : RUN cp /teensyduino/bin/arm-none-eabi-as /usr/bin/as
 ---> Using cache
 ---> e45a5c976edd
Step 35/37 : WORKDIR /teensyduino
 ---> Using cache
 ---> ba0349f757b5
Step 36/37 : RUN rm -rf ./${DOCKER_ARDUINO_VERSION} &&     rm TeensyduinoInstall.linux64
 ---> Using cache
 ---> 9b7626313339
Step 37/37 : CMD ["/bin/bash","/helper_scripts/entrypoint.sh"]
 ---> Using cache
 ---> c0e4e965732f
Successfully built c0e4e965732f
Successfully tagged teensy_dev:latest
cp: cannot stat '/libs/*': No such file or directory
/teensyduino/libraries
-- Using Teensy3 Core
-- /teensyduino/libraries
-- AccelStepper/src
-- Adafruit_nRF8001/utility
-- Audio/utility
-- Bounce2/src
-- CryptoAccel/src
-- Encoder/utility
-- Ethernet/src
-- Ethernet/src/utility
-- FNET/src
-- FastLED/src
-- FlexIO_t4/src
-- ILI9488_t3/src
-- IRremote/src
-- Keypad/src
-- LedControl/src
-- LiquidCrystal/src
-- LittleFS/src
-- MFRC522/src
-- MIDI/src
-- NXPMotionSense/utility
-- NativeEthernet/src
-- NativeEthernet/src/utility
-- PS2Keyboard/utility
-- ResponsiveAnalogRead/src
-- SD/src
-- SdFat/src
-- Snooze/src
-- USBHost_t36/utility
-- Wire/utility
-- ADC
-- AccelStepper
-- Adafruit-GFX-Library
-- Adafruit_BusIO
-- Adafruit_NeoPixel
-- Adafruit_STMPE610
-- Adafruit_VS1053
-- Adafruit_nRF8001
-- AltSoftSerial
-- Artnet
-- Audio
-- Bounce
-- Bounce2
-- CapacitiveSensor
-- CryptoAccel
-- DS1307RTC
-- DmxSimple
-- DogLcd
-- EEPROM
-- EasyTransfer
-- EasyTransferI2C
-- Encoder
-- Entropy
-- Ethernet
-- FNET
-- FastCRC
-- FastLED
-- FlexCAN
-- FlexCAN_T4
-- FlexIO_t4
-- FlexiTimer2
-- FreqCount
-- FreqMeasure
-- FreqMeasureMulti
-- FrequencyTimer2
-- ILI9341_t3
-- ILI9488_t3
-- IRremote
-- Keypad
-- LedControl
-- LedDisplay
-- LiquidCrystal
-- LiquidCrystalFast
-- LittleFS
-- LowPower
-- MFRC522
-- MIDI
-- Metro
-- MsTimer2
-- NXPMotionSense
-- NativeEthernet
-- OSC
-- OctoWS2811
-- OneWire
-- PS2Keyboard
-- PWMServo
-- Ping
-- PulsePosition
-- QuadEncoder
-- RA8875
-- RadioHead
-- ResponsiveAnalogRead
-- SD
-- SPI
-- SPIFlash
-- ST7735_t3
-- SdFat
-- SerialFlash
-- Servo
-- ShiftPWM
-- Snooze
-- SoftPWM
-- SoftwareSerial
-- TFT_ILI9163C
-- Talkie
-- TeensyThreads
-- Time
-- TimeAlarms
-- TimerOne
-- TimerThree
-- TinyGPS
-- Tlc5940
-- TouchScreen
-- USBHost_t36
-- UTFT
-- VirtualWire
-- WS2812Serial
-- Wire
-- XBee
-- XPT2046_Touchscreen
-- i2c_t3
-- ks0108
-- ssd1351
-- x10
-- /teensyduino/libraries/AccelStepper/src/teensyduino/libraries/Adafruit_nRF8001/utility/teensyduino/libraries/Audio/utility/teensyduino/libraries/Bounce2/src/teensyduino/libraries/CryptoAccel/src/teensyduino/libraries/Encoder/utility/teensyduino/libraries/Ethernet/src/teensyduino/libraries/Ethernet/src/utility/teensyduino/libraries/FNET/src/teensyduino/libraries/FastLED/src/teensyduino/libraries/FlexIO_t4/src/teensyduino/libraries/ILI9488_t3/src/teensyduino/libraries/IRremote/src/teensyduino/libraries/Keypad/src/teensyduino/libraries/LedControl/src/teensyduino/libraries/LiquidCrystal/src/teensyduino/libraries/LittleFS/src/teensyduino/libraries/MFRC522/src/teensyduino/libraries/MIDI/src/teensyduino/libraries/NXPMotionSense/utility/teensyduino/libraries/NativeEthernet/src/teensyduino/libraries/NativeEthernet/src/utility/teensyduino/libraries/PS2Keyboard/utility/teensyduino/libraries/ResponsiveAnalogRead/src/teensyduino/libraries/SD/src/teensyduino/libraries/SdFat/src/teensyduino/libraries/Snooze/src/teensyduino/libraries/USBHost_t36/utility/teensyduino/libraries/Wire/utilityADCAccelStepperAdafruit-GFX-LibraryAdafruit_BusIOAdafruit_NeoPixelAdafruit_STMPE610Adafruit_VS1053Adafruit_nRF8001AltSoftSerialArtnetAudioBounceBounce2CapacitiveSensorCryptoAccelDS1307RTCDmxSimpleDogLcdEEPROMEasyTransferEasyTransferI2CEncoderEntropyEthernetFNETFastCRCFastLEDFlexCANFlexCAN_T4FlexIO_t4FlexiTimer2FreqCountFreqMeasureFreqMeasureMultiFrequencyTimer2ILI9341_t3ILI9488_t3IRremoteKeypadLedControlLedDisplayLiquidCrystalLiquidCrystalFastLittleFSMFRC522MIDIMetroMsTimer2NXPMotionSenseOSCOctoWS2811OneWirePS2KeyboardPWMServoPingPulsePositionRA8875RadioHeadResponsiveAnalogReadSDSPISPIFlashST7735_t3SdFatSerialFlashServoShiftPWMSnoozeSoftPWMSoftwareSerialTFT_ILI9163CTalkieTeensyThreadsTimeTimeAlarmsTimerOneTimerThreeTinyGPSTlc5940TouchScreenUTFTVirtualWireWS2812SerialWireXBeeXPT2046_Touchscreeni2c_t3ssd1351x10
-- ADCAccelStepperAdafruit-GFX-LibraryAdafruit_BusIOAdafruit_NeoPixelAdafruit_STMPE610Adafruit_VS1053Adafruit_nRF8001AltSoftSerialArtnetAudioBounceBounce2CapacitiveSensorCryptoAccelDS1307RTCDmxSimpleDogLcdEEPROMEasyTransferEasyTransferI2CEncoderEntropyEthernetFNETFastCRCFastLEDFlexCANFlexCAN_T4FlexIO_t4FlexiTimer2FreqCountFreqMeasureFreqMeasureMultiFrequencyTimer2ILI9341_t3ILI9488_t3IRremoteKeypadLedControlLedDisplayLiquidCrystalLiquidCrystalFastLittleFSMFRC522MIDIMetroMsTimer2NXPMotionSenseOSCOctoWS2811OneWirePS2KeyboardPWMServoPingPulsePositionRA8875RadioHeadResponsiveAnalogReadSDSPISPIFlashST7735_t3SdFatSerialFlashServoShiftPWMSnoozeSoftPWMSoftwareSerialTFT_ILI9163CTalkieTeensyThreadsTimeTimeAlarmsTimerOneTimerThreeTinyGPSTlc5940TouchScreenUTFTVirtualWireWS2812SerialWireXBeeXPT2046_Touchscreeni2c_t3ssd1351x10
-- /teensyduino/libraries/AccelStepper/src/teensyduino/libraries/Adafruit_nRF8001/utility/teensyduino/libraries/Audio/utility/teensyduino/libraries/Bounce2/src/teensyduino/libraries/CryptoAccel/src/teensyduino/libraries/Encoder/utility/teensyduino/libraries/Ethernet/src/teensyduino/libraries/Ethernet/src/utility/teensyduino/libraries/FNET/src/teensyduino/libraries/FastLED/src/teensyduino/libraries/FlexIO_t4/src/teensyduino/libraries/ILI9488_t3/src/teensyduino/libraries/IRremote/src/teensyduino/libraries/Keypad/src/teensyduino/libraries/LedControl/src/teensyduino/libraries/LiquidCrystal/src/teensyduino/libraries/LittleFS/src/teensyduino/libraries/MFRC522/src/teensyduino/libraries/MIDI/src/teensyduino/libraries/NXPMotionSense/utility/teensyduino/libraries/NativeEthernet/src/teensyduino/libraries/NativeEthernet/src/utility/teensyduino/libraries/PS2Keyboard/utility/teensyduino/libraries/ResponsiveAnalogRead/src/teensyduino/libraries/SD/src/teensyduino/libraries/SdFat/src/teensyduino/libraries/Snooze/src/teensyduino/libraries/USBHost_t36/utility/teensyduino/libraries/Wire/utilityADCAccelStepperAdafruit-GFX-LibraryAdafruit_BusIOAdafruit_NeoPixelAdafruit_STMPE610Adafruit_VS1053Adafruit_nRF8001AltSoftSerialArtnetAudioBounceBounce2CapacitiveSensorCryptoAccelDS1307RTCDmxSimpleDogLcdEEPROMEasyTransferEasyTransferI2CEncoderEntropyEthernetFNETFastCRCFastLEDFlexCANFlexCAN_T4FlexIO_t4FlexiTimer2FreqCountFreqMeasureFreqMeasureMultiFrequencyTimer2ILI9341_t3ILI9488_t3IRremoteKeypadLedControlLedDisplayLiquidCrystalLiquidCrystalFastLittleFSMFRC522MIDIMetroMsTimer2NXPMotionSenseOSCOctoWS2811OneWirePS2KeyboardPWMServoPingPulsePositionRA8875RadioHeadResponsiveAnalogReadSDSPISPIFlashST7735_t3SdFatSerialFlashServoShiftPWMSnoozeSoftPWMSoftwareSerialTFT_ILI9163CTalkieTeensyThreadsTimeTimeAlarmsTimerOneTimerThreeTinyGPSTlc5940TouchScreenUTFTVirtualWireWS2812SerialWireXBeeXPT2046_Touchscreeni2c_t3ssd1351x10
-- Configuring done
-- Generating done
-- Build files have been written to: /teensyduino/build
[  1%] Built target x10
[  2%] Built target Adafruit_NeoPixel
[  3%] Built target AltSoftSerial
[  3%] Built target ADC
[  4%] Built target AccelStepper
[  5%] Built target Adafruit-GFX-Library
[  6%] Built target CapacitiveSensor
[  6%] Built target Adafruit_STMPE610
[  6%] Built target Artnet
[  7%] Built target Bounce
[  8%] Built target Adafruit_VS1053
[  9%] Built target Adafruit_nRF8001
[ 10%] Built target Adafruit_BusIO
[ 11%] Built target CryptoAccel
[ 11%] Built target Bounce2
[ 12%] Built target EEPROM
[ 12%] Built target DogLcd
[ 12%] Built target Encoder
[ 14%] Built target DS1307RTC
[ 14%] Built target Entropy
[ 14%] Built target DmxSimple
[ 14%] Built target FNET
[ 14%] Built target EasyTransfer
[ 14%] Built target FlexCAN_T4
[ 15%] Built target EasyTransferI2C
[ 18%] Built target Ethernet
[ 18%] Built target FastCRC
[ 19%] Building CXX object libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/src/FlexIOSPI.cpp.obj
[ 19%] Built target FlexCAN
[ 21%] Built target FastLED
[ 42%] Built target Audio
[ 42%] Building CXX object libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/src/FlexSerial.cpp.obj
[ 43%] Built target FrequencyTimer2
[ 43%] Built target FlexiTimer2
[ 44%] Built target FreqCount
In file included from /teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.h:29:0,
                 from /teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:1:
/teensyduino/libraries/FlexIO_t4/src/FlexIO_t4.h:35:2: error: #error "Sorry, Flex IO only works on Teensy 4.x boards"
 #error "Sorry, Flex IO only works on Teensy 4.x boards"
  ^
[ 44%] Built target FreqMeasure
[ 44%] Built target Keypad
[ 47%] Built target ILI9488_t3
[ 47%] Built target LedControl
[ 47%] Built target ILI9341_t3
[ 48%] Built target FreqMeasureMulti
[ 50%] Built target IRremote
[ 50%] Built target LiquidCrystal
[ 50%] Built target LedDisplay
In file included from /teensyduino/libraries/FlexIO_t4/src/FlexSerial.h:29:0,
                 from /teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:1:
/teensyduino/libraries/FlexIO_t4/src/FlexIO_t4.h:35:2: error: #error "Sorry, Flex IO only works on Teensy 4.x boards"
 #error "Sorry, Flex IO only works on Teensy 4.x boards"
  ^
[ 51%] Built target MIDI
[ 52%] Built target LittleFS
[ 52%] Built target LiquidCrystalFast
[ 52%] Built target Metro
[ 54%] Built target NXPMotionSense
[ 54%] Built target MFRC522
[ 56%] Built target OSC
[ 57%] Built target PulsePosition
[ 58%] Built target OneWire
[ 58%] Built target OctoWS2811
[ 58%] Built target Ping
[ 59%] Built target MsTimer2
[ 60%] Built target PWMServo
/teensyduino/libraries/FlexIO_t4/src/FlexIO_t4.h:89:2: error: 'IMXRT_FLEXIO_t' does not name a type
  IMXRT_FLEXIO_t & port() { return *(IMXRT_FLEXIO_t *)port_addr; }
  ^
[ 60%] Built target PS2Keyboard
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp: In member function 'bool FlexIOSPI::begin()':
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:76:2: error: 'IMXRT_FLEXIO_t' was not declared in this scope
  IMXRT_FLEXIO_t *p = &_pflex->port();
  ^
[ 60%] Built target RA8875
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:76:18: error: 'p' was not declared in this scope
  IMXRT_FLEXIO_t *p = &_pflex->port();
                  ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:76:31: error: 'class FlexIOHandler' has no member named 'port'
  IMXRT_FLEXIO_t *p = &_pflex->port();
                               ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:110:29: error: 'FLEXIO_SHIFTCTL_TIMPOL' was not declared in this scope
  p->SHIFTCTL[_tx_shifter] = FLEXIO_SHIFTCTL_TIMPOL | FLEXIO_SHIFTCTL_PINCFG(3) | FLEXIO_SHIFTCTL_SMOD(2) |
                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:110:78: error: 'FLEXIO_SHIFTCTL_PINCFG' was not declared in this scope
  p->SHIFTCTL[_tx_shifter] = FLEXIO_SHIFTCTL_TIMPOL | FLEXIO_SHIFTCTL_PINCFG(3) | FLEXIO_SHIFTCTL_SMOD(2) |
                                                                              ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:110:104: error: 'FLEXIO_SHIFTCTL_SMOD' was not declared in this scope
  p->SHIFTCTL[_tx_shifter] = FLEXIO_SHIFTCTL_TIMPOL | FLEXIO_SHIFTCTL_PINCFG(3) | FLEXIO_SHIFTCTL_SMOD(2) |
                                                                                                        ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:111:61: error: 'FLEXIO_SHIFTCTL_TIMSEL' was not declared in this scope
                                FLEXIO_SHIFTCTL_TIMSEL(_timer) | FLEXIO_SHIFTCTL_PINSEL(_mosi_flex_pin); // 0x0003_0002;
                                                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:111:102: error: 'FLEXIO_SHIFTCTL_PINSEL' was not declared in this scope
                                FLEXIO_SHIFTCTL_TIMSEL(_timer) | FLEXIO_SHIFTCTL_PINSEL(_mosi_flex_pin); // 0x0003_0002;
                                                                                                      ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:118:45: error: 'FLEXIO_TIMCTL_TRGSEL' was not declared in this scope
  p->TIMCTL[_timer] =  FLEXIO_TIMCTL_TRGSEL(1) | FLEXIO_TIMCTL_TRGPOL | FLEXIO_TIMCTL_TRGSRC 
                                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:118:49: error: 'FLEXIO_TIMCTL_TRGPOL' was not declared in this scope
  p->TIMCTL[_timer] =  FLEXIO_TIMCTL_TRGSEL(1) | FLEXIO_TIMCTL_TRGPOL | FLEXIO_TIMCTL_TRGSRC 
                                                 ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:118:72: error: 'FLEXIO_TIMCTL_TRGSRC' was not declared in this scope
  p->TIMCTL[_timer] =  FLEXIO_TIMCTL_TRGSEL(1) | FLEXIO_TIMCTL_TRGPOL | FLEXIO_TIMCTL_TRGSRC 
                                                                        ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:119:30: error: 'FLEXIO_TIMCTL_PINCFG' was not declared in this scope
      | FLEXIO_TIMCTL_PINCFG(3) | FLEXIO_TIMCTL_PINSEL(_sck_flex_pin)| FLEXIO_TIMCTL_TIMOD(1);  // 0x01C0_0001;
                              ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:119:68: error: 'FLEXIO_TIMCTL_PINSEL' was not declared in this scope
      | FLEXIO_TIMCTL_PINCFG(3) | FLEXIO_TIMCTL_PINSEL(_sck_flex_pin)| FLEXIO_TIMCTL_TIMOD(1);  // 0x01C0_0001;
                                                                    ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:119:92: error: 'FLEXIO_TIMCTL_TIMOD' was not declared in this scope
      | FLEXIO_TIMCTL_PINCFG(3) | FLEXIO_TIMCTL_PINSEL(_sck_flex_pin)| FLEXIO_TIMCTL_TIMOD(1);  // 0x01C0_0001;
                                                                                            ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:123:45: error: 'FLEXIO_TIMCFG_TIMOUT' was not declared in this scope
   p->TIMCFG[_timer] = FLEXIO_TIMCFG_TIMOUT(1) | FLEXIO_TIMCFG_TIMDIS(2) | FLEXIO_TIMCFG_TIMENA(2) 
                                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:123:71: error: 'FLEXIO_TIMCFG_TIMDIS' was not declared in this scope
   p->TIMCFG[_timer] = FLEXIO_TIMCFG_TIMOUT(1) | FLEXIO_TIMCFG_TIMDIS(2) | FLEXIO_TIMCFG_TIMENA(2) 
                                                                       ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:123:97: error: 'FLEXIO_TIMCFG_TIMENA' was not declared in this scope
   p->TIMCFG[_timer] = FLEXIO_TIMCFG_TIMOUT(1) | FLEXIO_TIMCFG_TIMDIS(2) | FLEXIO_TIMCFG_TIMENA(2) 
                                                                                                 ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:124:52: error: 'FLEXIO_TIMCFG_TSTOP' was not declared in this scope
                             | FLEXIO_TIMCFG_TSTOP(2) | FLEXIO_TIMCFG_TSTART; //0x0100_2222;
                                                    ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:124:56: error: 'FLEXIO_TIMCFG_TSTART' was not declared in this scope
                             | FLEXIO_TIMCFG_TSTOP(2) | FLEXIO_TIMCFG_TSTART; //0x0100_2222;
                                                        ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:131:7: error: 'FLEXIO_TIMCTL_PINPOL' was not declared in this scope
       FLEXIO_TIMCTL_PINPOL | FLEXIO_TIMCTL_TIMOD(3);  // 0003_0383;
       ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:134:45: error: 'FLEXIO_TIMCFG_TIMOUT' was not declared in this scope
   p->TIMCFG[_timer] = FLEXIO_TIMCFG_TIMOUT(1) | FLEXIO_TIMCFG_TIMDIS(2) | FLEXIO_TIMCFG_TIMENA(2); 
                                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:134:71: error: 'FLEXIO_TIMCFG_TIMDIS' was not declared in this scope
   p->TIMCFG[_timer] = FLEXIO_TIMCFG_TIMOUT(1) | FLEXIO_TIMCFG_TIMDIS(2) | FLEXIO_TIMCFG_TIMENA(2); 
                                                                       ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:134:97: error: 'FLEXIO_TIMCFG_TIMENA' was not declared in this scope
   p->TIMCFG[_timer] = FLEXIO_TIMCFG_TIMOUT(1) | FLEXIO_TIMCFG_TIMDIS(2) | FLEXIO_TIMCFG_TIMENA(2); 
                                                                                                 ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:138:12: error: 'FLEXIO_CTRL_FLEXEN' was not declared in this scope
  p->CTRL = FLEXIO_CTRL_FLEXEN;
            ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:146:36: error: 'IOMUXC_PAD_DSE' was not declared in this scope
  uint32_t fastio = IOMUXC_PAD_DSE(7) | IOMUXC_PAD_SPEED(2);
                                    ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:146:58: error: 'IOMUXC_PAD_SPEED' was not declared in this scope
  uint32_t fastio = IOMUXC_PAD_DSE(7) | IOMUXC_PAD_SPEED(2);
                                                          ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:150:32: error: 'portControlRegister' was not declared in this scope
  *(portControlRegister(_mosiPin)) = fastio;
                                ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:152:46: error: 'IOMUXC_PAD_PUE' was not declared in this scope
  *(portControlRegister(_misoPin)) = fastio | IOMUXC_PAD_PUE | IOMUXC_PAD_PUS(3);  // maybe add our own internal PU?
                                              ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:152:79: error: 'IOMUXC_PAD_PUS' was not declared in this scope
  *(portControlRegister(_misoPin)) = fastio | IOMUXC_PAD_PUE | IOMUXC_PAD_PUS(3);  // maybe add our own internal PU?
                                                                               ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:162:29: error: 'class FlexIOHandler' has no member named 'port'
  _shiftBufOutReg = &_pflex->port().SHIFTBUFBBS[_tx_shifter];
                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:163:28: error: 'class FlexIOHandler' has no member named 'port'
  _shiftBufInReg = &_pflex->port().SHIFTBUFBIS[_rx_shifter];;
                            ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp: In member function 'void FlexIOSPI::beginTransaction(FlexIOSPISettings)':
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:230:11: error: 'class FlexIOHandler' has no member named 'port'
   _pflex->port().TIMCMP[_timer] = div | (_nTransferBits*2-1)<<8; // Set the clk div for shifter and set transfer length
           ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp: In member function 'void FlexIOSPI::setShiftBufferOut(uint32_t, uint8_t)':
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:257:11: error: 'class FlexIOHandler' has no member named 'port'
   _pflex->port().SHIFTBUFBIS[_tx_shifter] = val << (32-nbits);
           ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:259:11: error: 'class FlexIOHandler' has no member named 'port'
   _pflex->port().SHIFTBUF[_tx_shifter] = val;
           ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp: In member function 'uint32_t FlexIOSPI::getShiftBufferIn(uint8_t)':
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:288:21: error: 'class FlexIOHandler' has no member named 'port'
   ret_val = _pflex->port().SHIFTBUFBIS[_rx_shifter]; 
                     ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:290:21: error: 'class FlexIOHandler' has no member named 'port'
   ret_val = _pflex->port().SHIFTBUF[_rx_shifter] >> (32-nbits);
                     ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp: In member function 'uint32_t FlexIOSPI::transferNBits(uint32_t, uint8_t)':
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:322:33: error: 'class FlexIOHandler' has no member named 'port'
  uint16_t timcmp_save = _pflex->port().TIMCMP[_timer]; // remember value coming in
                                 ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:323:10: error: 'class FlexIOHandler' has no member named 'port'
  _pflex->port().TIMCMP[_timer] = (timcmp_save & 0xff) | (nbits*2-1)<<8; // Adjust transmission length to nbits
          ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:328:14: error: 'class FlexIOHandler' has no member named 'port'
  if (_pflex->port().SHIFTSTAT & _rx_shifter_mask) {
              ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:335:19: error: 'class FlexIOHandler' has no member named 'port'
  while (!(_pflex->port().SHIFTSTAT & _rx_shifter_mask) && (--timeout)) ;
                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:337:14: error: 'class FlexIOHandler' has no member named 'port'
  if (_pflex->port().SHIFTSTAT & _rx_shifter_mask) {
              ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:341:10: error: 'class FlexIOHandler' has no member named 'port'
  _pflex->port().TIMCMP[_timer] = timcmp_save;
          ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp: In member function 'void FlexIOSPI::transferBufferNBits(const void*, void*, size_t, uint8_t)':
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:358:10: error: 'class FlexIOHandler' has no member named 'port'
  _pflex->port().SHIFTERR = _rx_shifter_mask | _tx_shifter_mask; // clear out any previous errors
          ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:359:19: error: 'class FlexIOHandler' has no member named 'port'
  while (!(_pflex->port().SHIFTSTAT & _tx_shifter_mask))  ; // wait for room for the first character
                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:368:20: error: 'class FlexIOHandler' has no member named 'port'
   while (!(_pflex->port().SHIFTSTAT & _tx_shifter_mask))  ;
                    ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:377:21: error: 'class FlexIOHandler' has no member named 'port'
   while  (!(_pflex->port().SHIFTSTAT & _rx_shifter_mask)) ;
                     ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:386:20: error: 'class FlexIOHandler' has no member named 'port'
  while  (!(_pflex->port().SHIFTSTAT & _rx_shifter_mask) && !(_pflex->port().SHIFTERR & _rx_shifter_mask)) ;
                    ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:386:70: error: 'class FlexIOHandler' has no member named 'port'
  while  (!(_pflex->port().SHIFTSTAT & _rx_shifter_mask) && !(_pflex->port().SHIFTERR & _rx_shifter_mask)) ;
                                                                      ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp: In member function 'bool FlexIOSPI::transfer(const void*, void*, size_t, EventResponderRef)':
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:532:10: error: 'class FlexIOHandler' has no member named 'port'
  _pflex->port().SHIFTSDEN |= _rx_shifter_mask | _tx_shifter_mask;
          ^
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp: In member function 'void FlexIOSPI::dma_rxisr()':
/teensyduino/libraries/FlexIO_t4/src/FlexIOSPI.cpp:573:11: error: 'class FlexIOHandler' has no member named 'port'
   _pflex->port().SHIFTSDEN &= ~(_rx_shifter_mask | _tx_shifter_mask);  // turn off DMA on both RX and TX
           ^
make[2]: *** [libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/build.make:63: libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/src/FlexIOSPI.cpp.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
/teensyduino/libraries/FlexIO_t4/src/FlexIO_t4.h:89:2: error: 'IMXRT_FLEXIO_t' does not name a type
  IMXRT_FLEXIO_t & port() { return *(IMXRT_FLEXIO_t *)port_addr; }
  ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp: In member function 'bool FlexSerial::begin(uint32_t, bool)':
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:47:3: error: 'IMXRT_FLEXIO_t' was not declared in this scope
   IMXRT_FLEXIO_t *p = &_tx_pflex->port();
   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:47:19: error: 'p' was not declared in this scope
   IMXRT_FLEXIO_t *p = &_tx_pflex->port();
                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:47:35: error: 'class FlexIOHandler' has no member named 'port'
   IMXRT_FLEXIO_t *p = &_tx_pflex->port();
                                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:97:53: error: 'FLEXIO_SHIFTCFG_SSTOP' was not declared in this scope
   p->SHIFTCFG[_tx_shifter] = FLEXIO_SHIFTCFG_SSTOP(3) | FLEXIO_SHIFTCFG_SSTART(2); //0x0000_0032;
                                                     ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:97:81: error: 'FLEXIO_SHIFTCFG_SSTART' was not declared in this scope
   p->SHIFTCFG[_tx_shifter] = FLEXIO_SHIFTCFG_SSTOP(3) | FLEXIO_SHIFTCFG_SSTART(2); //0x0000_0032;
                                                                                 ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:98:54: error: 'FLEXIO_SHIFTCTL_PINCFG' was not declared in this scope
   p->SHIFTCTL[_tx_shifter] = FLEXIO_SHIFTCTL_PINCFG(3) | FLEXIO_SHIFTCTL_SMOD(2) |
                                                      ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:98:80: error: 'FLEXIO_SHIFTCTL_SMOD' was not declared in this scope
   p->SHIFTCTL[_tx_shifter] = FLEXIO_SHIFTCTL_PINCFG(3) | FLEXIO_SHIFTCTL_SMOD(2) |
                                                                                ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:99:65: error: 'FLEXIO_SHIFTCTL_TIMSEL' was not declared in this scope
                                 FLEXIO_SHIFTCTL_TIMSEL(_tx_timer) | FLEXIO_SHIFTCTL_PINSEL(_tx_flex_pin); // 0x0003_0002;
                                                                 ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:99:104: error: 'FLEXIO_SHIFTCTL_PINSEL' was not declared in this scope
                                 FLEXIO_SHIFTCTL_TIMSEL(_tx_timer) | FLEXIO_SHIFTCTL_PINSEL(_tx_flex_pin); // 0x0003_0002;
                                                                                                        ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:101:26: error: 'FLEXIO_TIMCFG_TSTART' was not declared in this scope
   p->TIMCFG[_tx_timer] = FLEXIO_TIMCFG_TSTART | FLEXIO_TIMCFG_TSTOP(2) |
                          ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:101:70: error: 'FLEXIO_TIMCFG_TSTOP' was not declared in this scope
   p->TIMCFG[_tx_timer] = FLEXIO_TIMCFG_TSTART | FLEXIO_TIMCFG_TSTOP(2) |
                                                                      ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:102:51: error: 'FLEXIO_TIMCFG_TIMENA' was not declared in this scope
                             FLEXIO_TIMCFG_TIMENA(2) |  FLEXIO_TIMCFG_TIMDIS(2); //0x0000_2222;
                                                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:102:78: error: 'FLEXIO_TIMCFG_TIMDIS' was not declared in this scope
                             FLEXIO_TIMCFG_TIMENA(2) |  FLEXIO_TIMCFG_TIMDIS(2); //0x0000_2222;
                                                                              ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:103:47: error: 'FLEXIO_TIMCTL_TIMOD' was not declared in this scope
   p->TIMCTL[_tx_timer] = FLEXIO_TIMCTL_TIMOD(1) | FLEXIO_TIMCTL_TRGPOL | FLEXIO_TIMCTL_TRGSRC
                                               ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:103:51: error: 'FLEXIO_TIMCTL_TRGPOL' was not declared in this scope
   p->TIMCTL[_tx_timer] = FLEXIO_TIMCTL_TIMOD(1) | FLEXIO_TIMCTL_TRGPOL | FLEXIO_TIMCTL_TRGSRC
                                                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:103:74: error: 'FLEXIO_TIMCTL_TRGSRC' was not declared in this scope
   p->TIMCTL[_tx_timer] = FLEXIO_TIMCTL_TIMOD(1) | FLEXIO_TIMCTL_TRGPOL | FLEXIO_TIMCTL_TRGSRC
                                                                          ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:104:69: error: 'FLEXIO_TIMCTL_TRGSEL' was not declared in this scope
                             | FLEXIO_TIMCTL_TRGSEL(4*_tx_shifter + 1) | FLEXIO_TIMCTL_PINSEL(_tx_flex_pin);  // 0x01C0_0001;
                                                                     ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:104:106: error: 'FLEXIO_TIMCTL_PINSEL' was not declared in this scope
                             | FLEXIO_TIMCTL_TRGSEL(4*_tx_shifter + 1) | FLEXIO_TIMCTL_PINSEL(_tx_flex_pin);  // 0x01C0_0001;
                                                                                                          ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:107:13: error: 'FLEXIO_CTRL_FLEXEN' was not declared in this scope
   p->CTRL = FLEXIO_CTRL_FLEXEN;
             ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:157:3: error: 'IMXRT_FLEXIO_t' was not declared in this scope
   IMXRT_FLEXIO_t *p = &_rx_pflex->port();
   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:157:19: error: 'p' was not declared in this scope
   IMXRT_FLEXIO_t *p = &_rx_pflex->port();
                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:157:35: error: 'class FlexIOHandler' has no member named 'port'
   IMXRT_FLEXIO_t *p = &_rx_pflex->port();
                                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:207:53: error: 'FLEXIO_SHIFTCFG_SSTOP' was not declared in this scope
   p->SHIFTCFG[_rx_shifter] = FLEXIO_SHIFTCFG_SSTOP(3) | FLEXIO_SHIFTCFG_SSTART(2); //0x0000_0032;
                                                     ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:207:81: error: 'FLEXIO_SHIFTCFG_SSTART' was not declared in this scope
   p->SHIFTCFG[_rx_shifter] = FLEXIO_SHIFTCFG_SSTOP(3) | FLEXIO_SHIFTCFG_SSTART(2); //0x0000_0032;
                                                                                 ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:208:30: error: 'FLEXIO_SHIFTCTL_TIMPOL' was not declared in this scope
   p->SHIFTCTL[_rx_shifter] = FLEXIO_SHIFTCTL_TIMPOL | FLEXIO_SHIFTCTL_SMOD(1) |
                              ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:208:77: error: 'FLEXIO_SHIFTCTL_SMOD' was not declared in this scope
   p->SHIFTCTL[_rx_shifter] = FLEXIO_SHIFTCTL_TIMPOL | FLEXIO_SHIFTCTL_SMOD(1) |
                                                                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:209:65: error: 'FLEXIO_SHIFTCTL_TIMSEL' was not declared in this scope
                                 FLEXIO_SHIFTCTL_TIMSEL(_rx_timer) | FLEXIO_SHIFTCTL_PINSEL(_rx_flex_pin); // 0x0080_0001;
                                                                 ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:209:104: error: 'FLEXIO_SHIFTCTL_PINSEL' was not declared in this scope
                                 FLEXIO_SHIFTCTL_TIMSEL(_rx_timer) | FLEXIO_SHIFTCTL_PINSEL(_rx_flex_pin); // 0x0080_0001;
                                                                                                        ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:212:26: error: 'FLEXIO_TIMCFG_TSTART' was not declared in this scope
   p->TIMCFG[_rx_timer] = FLEXIO_TIMCFG_TSTART | FLEXIO_TIMCFG_TSTOP(2) |
                          ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:212:70: error: 'FLEXIO_TIMCFG_TSTOP' was not declared in this scope
   p->TIMCFG[_rx_timer] = FLEXIO_TIMCFG_TSTART | FLEXIO_TIMCFG_TSTOP(2) |
                                                                      ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:213:51: error: 'FLEXIO_TIMCFG_TIMENA' was not declared in this scope
                             FLEXIO_TIMCFG_TIMENA(4) | FLEXIO_TIMCFG_TIMDIS(2) |
                                                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:213:77: error: 'FLEXIO_TIMCFG_TIMDIS' was not declared in this scope
                             FLEXIO_TIMCFG_TIMENA(4) | FLEXIO_TIMCFG_TIMDIS(2) |
                                                                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:214:51: error: 'FLEXIO_TIMCFG_TIMRST' was not declared in this scope
                             FLEXIO_TIMCFG_TIMRST(4) | FLEXIO_TIMCFG_TIMOUT(2); //0x204_2422
                                                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:214:77: error: 'FLEXIO_TIMCFG_TIMOUT' was not declared in this scope
                             FLEXIO_TIMCFG_TIMRST(4) | FLEXIO_TIMCFG_TIMOUT(2); //0x204_2422
                                                                             ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:216:47: error: 'FLEXIO_TIMCTL_TIMOD' was not declared in this scope
   p->TIMCTL[_rx_timer] = FLEXIO_TIMCTL_TIMOD(1) | FLEXIO_TIMCTL_PINPOL | FLEXIO_TIMCTL_PINSEL(_rx_flex_pin);;  // 0x0000_0081;
                                               ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:216:51: error: 'FLEXIO_TIMCTL_PINPOL' was not declared in this scope
   p->TIMCTL[_rx_timer] = FLEXIO_TIMCTL_TIMOD(1) | FLEXIO_TIMCTL_PINPOL | FLEXIO_TIMCTL_PINSEL(_rx_flex_pin);;  // 0x0000_0081;
                                                   ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:216:107: error: 'FLEXIO_TIMCTL_PINSEL' was not declared in this scope
   p->TIMCTL[_rx_timer] = FLEXIO_TIMCTL_TIMOD(1) | FLEXIO_TIMCTL_PINPOL | FLEXIO_TIMCTL_PINSEL(_rx_flex_pin);;  // 0x0000_0081;
                                                                                                           ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:217:13: error: 'FLEXIO_CTRL_FLEXEN' was not declared in this scope
   p->CTRL = FLEXIO_CTRL_FLEXEN;         // make sure it is enabled. 
             ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp: In member function 'virtual size_t FlexSerial::write(uint8_t)':
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:306:13: error: 'class FlexIOHandler' has no member named 'port'
  _tx_pflex->port().SHIFTSIEN |= _tx_shifter_mask;  // enable interrupt on this one...
             ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:307:13: error: 'class FlexIOHandler' has no member named 'port'
  _tx_pflex->port().TIMIEN &= ~_tx_timer_mask; // Remove any timer interrupts
             ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:308:13: error: 'class FlexIOHandler' has no member named 'port'
  _tx_pflex->port().TIMSTAT = _tx_timer_mask;  // Clear the state. 
             ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp: In member function 'virtual bool FlexSerial::call_back(FlexIOHandler*)':
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:364:2: error: 'IMXRT_FLEXIO_t' was not declared in this scope
  IMXRT_FLEXIO_t *p = &pflex->port();
  ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:364:18: error: 'p' was not declared in this scope
  IMXRT_FLEXIO_t *p = &pflex->port();
                  ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:364:30: error: 'class FlexIOHandler' has no member named 'port'
  IMXRT_FLEXIO_t *p = &pflex->port();
                              ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:366:18: error: 'class FlexIOHandler' has no member named 'port'
   if (_rx_pflex->port().SHIFTSTAT & _rx_shifter_mask) {
                  ^
/teensyduino/libraries/FlexIO_t4/src/FlexSerial.cpp:368:27: error: 'class FlexIOHandler' has no member named 'port'
    uint8_t c = _rx_pflex->port().SHIFTBUFBYS[_rx_shifter] & 0xff;
                           ^
[ 61%] Built target ResponsiveAnalogRead
[ 62%] Built target SPI
[ 62%] Built target SD
[ 62%] Built target SPIFlash
[ 63%] Built target ST7735_t3
make[2]: *** [libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/build.make:89: libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/src/FlexSerial.cpp.obj] Error 1
[ 64%] Built target ShiftPWM
[ 65%] Built target SdFat
[ 65%] Built target Servo
[ 72%] Built target RadioHead
make[1]: *** [CMakeFiles/Makefile2:3007: libraries/FlexIO_t4/CMakeFiles/FlexIO_t4.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 73%] Built target SerialFlash
[ 73%] Built target SoftwareSerial
[ 73%] Built target Snooze
[ 75%] Built target TeensyThreads
[ 75%] Built target SoftPWM
[ 76%] Built target TFT_ILI9163C
[ 77%] Built target Time
[ 77%] Built target Talkie
make: *** [Makefile:84: all] Error 2

```

--- a/config.json
+++ b/config.json
@@ -97,6 +97,13 @@
         ],
         "conflicts": []
     },
+    "FlexIO_t4": {
+        "supported": [
+            "40",
+            "41"
+        ],
+        "conflicts": []
+    },
     "FrequencyTimer2": {
         "supported": [
             "30",

